diff --git a/android/src/main/java/com/expensify/wallet/Utils.kt b/android/src/main/java/com/expensify/wallet/Utils.kt
index 34a764630606d24723bf5160cb9d81ab8ddd8443..6bb09ef4360657d32a2a443462db80a2f27a7d73 100644
--- a/android/src/main/java/com/expensify/wallet/Utils.kt
+++ b/android/src/main/java/com/expensify/wallet/Utils.kt
@@ -56,44 +56,44 @@ object Utils {
               )
             }
 
-            override fun reject(code: String?, userInfo: WritableMap) {
-              val errorMessage = "Error: ${code ?: "Unknown code"}\nUserInfo: $userInfo"
+            override fun reject(code: String, userInfo: WritableMap) {
+              val errorMessage = "Error: $code\nUserInfo: $userInfo"
               continuation.resumeWithException(
                 Exception(errorMessage)
               )
             }
 
-            override fun reject(code: String?, message: String?) {
-              val errorMessage = "Error: ${code ?: "Unknown code"}\nMessage: ${message ?: "No message provided"}"
+            override fun reject(code: String, message: String?) {
+              val errorMessage = "Error: $code\nMessage: ${message ?: "No message provided"}"
               continuation.resumeWithException(
                 Exception(errorMessage)
               )
             }
 
-            override fun reject(code: String?, message: String?, userInfo: WritableMap) {
+            override fun reject(code: String, message: String?, userInfo: WritableMap) {
               val errorMessage =
-                "Error: ${code ?: "Unknown code"}\nMessage: ${message ?: "No message provided"}\nUserInfo: $userInfo"
+                "Error: $code\nMessage: ${message ?: "No message provided"}\nUserInfo: $userInfo"
               continuation.resumeWithException(
                 Exception(errorMessage)
               )
             }
 
-            override fun reject(code: String?, message: String?, throwable: Throwable?) {
-              val errorMessage = "Error: ${code ?: "Unknown code"}\nMessage: ${message ?: "No message provided"}"
+            override fun reject(code: String, message: String?, throwable: Throwable?) {
+              val errorMessage = "Error: $code\nMessage: ${message ?: "No message provided"}"
               continuation.resumeWithException(
                 throwable ?: Exception(errorMessage)
               )
             }
 
-            override fun reject(code: String?, throwable: Throwable?) {
-              val errorMessage = "Error: ${code ?: "Unknown code"}"
+            override fun reject(code: String, throwable: Throwable?) {
+              val errorMessage = "Error: $code"
               continuation.resumeWithException(
                 throwable ?: Exception(errorMessage)
               )
             }
 
-            override fun reject(code: String?, throwable: Throwable?, userInfo: WritableMap) {
-              val errorMessage = "Error: ${code ?: "Unknown code"}\nUserInfo: $userInfo"
+            override fun reject(code: String, throwable: Throwable?, userInfo: WritableMap) {
+              val errorMessage = "Error: $code\nUserInfo: $userInfo"
               continuation.resumeWithException(
                 throwable ?: Exception(errorMessage)
               )
