diff --git a/dist/TokenBalancesController.cjs b/dist/TokenBalancesController.cjs
index 4216f3b2e86b2d837fd3856826bb531d4f401493..57e6eb184f16f01f6a330419d727b42880d8ab9a 100644
--- a/dist/TokenBalancesController.cjs
+++ b/dist/TokenBalancesController.cjs
@@ -354,7 +354,7 @@ class TokenBalancesController extends (0, polling_controller_1.StaticIntervalPol
             }
         }
         // Determine which accounts to process
-        const accountsToProcess = __classPrivateFieldGet(this, _TokenBalancesController_queryAllAccounts, "f")
+        const accountsToProcess = (queryAllAccounts ?? __classPrivateFieldGet(this, _TokenBalancesController_queryAllAccounts, "f"))
             ? allAccounts.map((a) => a.address)
             : [selected];
         const prev = this.state;
diff --git a/dist/TokenBalancesController.mjs b/dist/TokenBalancesController.mjs
index e458820ef2e026b2d5978279381c1da463e91fc0..440c4abd2e2cc6840a0b3b83f8537639bb86641a 100644
--- a/dist/TokenBalancesController.mjs
+++ b/dist/TokenBalancesController.mjs
@@ -352,7 +352,7 @@ export class TokenBalancesController extends StaticIntervalPollingController() {
             }
         }
         // Determine which accounts to process
-        const accountsToProcess = __classPrivateFieldGet(this, _TokenBalancesController_queryAllAccounts, "f")
+        const accountsToProcess = (queryAllAccounts ?? __classPrivateFieldGet(this, _TokenBalancesController_queryAllAccounts, "f"))
             ? allAccounts.map((a) => a.address)
             : [selected];
         const prev = this.state;
