diff --git a/dist/constants/bridge.cjs b/dist/constants/bridge.cjs
index 2ea1286fa1c8cdf6a318453f20f6b72eaf8e32b6..4ef2641abb90348ec935f03a68bd46aa7f248930 100644
--- a/dist/constants/bridge.cjs
+++ b/dist/constants/bridge.cjs
@@ -18,6 +18,7 @@ exports.ALLOWED_BRIDGE_CHAIN_IDS = [
     chains_1.CHAIN_IDS.MONAD,
     chains_1.CHAIN_IDS.HYPEREVM,
     chains_1.CHAIN_IDS.MEGAETH,
+    chains_1.CHAIN_IDS.ARC,
     keyring_api_1.SolScope.Mainnet,
     keyring_api_1.BtcScope.Mainnet,
     keyring_api_1.TrxScope.Mainnet,
@@ -54,6 +55,7 @@ exports.DEFAULT_CHAIN_RANKING = [
     { chainId: 'eip155:1329', name: 'Sei' },
     { chainId: 'eip155:999', name: 'HyperEVM' },
     { chainId: 'eip155:4326', name: 'MegaETH' },
+    { chainId: 'eip155:5042', name: 'Arc' },
     { chainId: 'eip155:324', name: 'zkSync' },
 ];
 exports.DEFAULT_FEATURE_FLAG_CONFIG = {
diff --git a/dist/constants/chains.cjs b/dist/constants/chains.cjs
index 842508af67e7422febd3badcc7a43242f5ecc806..32f95c05b7993301ceb68e3b83fa16773fe32d16 100644
--- a/dist/constants/chains.cjs
+++ b/dist/constants/chains.cjs
@@ -80,6 +80,7 @@ exports.CHAIN_IDS = {
     MONAD: '0x8f',
     HYPEREVM: '0x3e7',
     MEGAETH: '0x10e6',
+    ARC: '0x13b2',
 };
 exports.NETWORK_TYPES = {
     GOERLI: 'goerli',
@@ -131,6 +132,7 @@ exports.SEI_DISPLAY_NAME = 'Sei Network';
 exports.MONAD_DISPLAY_NAME = 'Monad';
 exports.HYPEREVM_DISPLAY_NAME = 'HyperEVM';
 exports.MEGAETH_DISPLAY_NAME = 'MegaETH';
+exports.ARC_DISPLAY_NAME = 'Arc';
 exports.NETWORK_TO_NAME_MAP = {
     [exports.NETWORK_TYPES.GOERLI]: exports.GOERLI_DISPLAY_NAME,
     [exports.NETWORK_TYPES.MAINNET]: exports.MAINNET_DISPLAY_NAME,
@@ -164,5 +166,6 @@ exports.NETWORK_TO_NAME_MAP = {
     [exports.CHAIN_IDS.MONAD]: exports.MONAD_DISPLAY_NAME,
     [exports.CHAIN_IDS.HYPEREVM]: exports.HYPEREVM_DISPLAY_NAME,
     [exports.CHAIN_IDS.MEGAETH]: exports.MEGAETH_DISPLAY_NAME,
+    [exports.CHAIN_IDS.ARC]: exports.ARC_DISPLAY_NAME,
 };
 //# sourceMappingURL=chains.cjs.map
\ No newline at end of file
diff --git a/dist/constants/tokens.cjs b/dist/constants/tokens.cjs
index 57f207adbd3a0323a50c5288a2971ad785c29501..0fc3584797940ff37e7fad3ae7882df18b7936eb 100644
--- a/dist/constants/tokens.cjs
+++ b/dist/constants/tokens.cjs
@@ -34,6 +34,7 @@ const CURRENCY_SYMBOLS = {
     MON: 'MON',
     HYPE: 'HYPE',
     MEGAETH: 'ETH',
+    ARC: 'USDC',
 };
 const ETH_SWAPS_TOKEN_OBJECT = {
     symbol: CURRENCY_SYMBOLS.ETH,
@@ -144,6 +145,14 @@ const HYPEREVM_SWAPS_TOKEN_OBJECT = {
 const MEGAETH_SWAPS_TOKEN_OBJECT = {
     ...ETH_SWAPS_TOKEN_OBJECT,
 };
+// Leaving for code consistency but we won't display it in the asset picker
+const ARC_SWAPS_TOKEN_OBJECT = {
+    symbol: 'USDC-native',
+    name: 'USDC-native',
+    address: '0x0000000000000000000000000000000000000000',
+    decimals: 18,
+    iconUrl: '',
+};
 exports.SWAPS_CHAINID_DEFAULT_TOKEN_MAP = {
     [chains_1.CHAIN_IDS.MAINNET]: ETH_SWAPS_TOKEN_OBJECT,
     [chains_1.CHAIN_IDS.LOCALHOST]: TEST_ETH_SWAPS_TOKEN_OBJECT,
@@ -161,6 +170,7 @@ exports.SWAPS_CHAINID_DEFAULT_TOKEN_MAP = {
     [chains_1.CHAIN_IDS.MONAD]: MONAD_SWAPS_TOKEN_OBJECT,
     [chains_1.CHAIN_IDS.HYPEREVM]: HYPEREVM_SWAPS_TOKEN_OBJECT,
     [chains_1.CHAIN_IDS.MEGAETH]: MEGAETH_SWAPS_TOKEN_OBJECT,
+    [chains_1.CHAIN_IDS.ARC]: ARC_SWAPS_TOKEN_OBJECT,
     [keyring_api_1.SolScope.Mainnet]: SOLANA_SWAPS_TOKEN_OBJECT,
     [keyring_api_1.SolScope.Devnet]: SOLANA_SWAPS_TOKEN_OBJECT,
     [keyring_api_1.BtcScope.Mainnet]: BTC_SWAPS_TOKEN_OBJECT,
@@ -182,5 +192,6 @@ exports.SYMBOL_TO_SLIP44_MAP = {
     TRX: 'slip44:195',
     MON: 'slip44:268435779',
     HYPE: 'slip44:2457',
+    'USDC-native': 'erc20:0x0000000000000000000000000000000000000000',
 };
 //# sourceMappingURL=tokens.cjs.map
\ No newline at end of file
diff --git a/dist/types.cjs b/dist/types.cjs
index 945d50d380f98144e12d308e331f0a25a9ac054c..5160b927ead302db4470ad06496e88e384605cb6 100644
--- a/dist/types.cjs
+++ b/dist/types.cjs
@@ -52,6 +52,7 @@ var ChainId;
     ChainId[ChainId["MONAD"] = 143] = "MONAD";
     ChainId[ChainId["HYPEREVM"] = 999] = "HYPEREVM";
     ChainId[ChainId["MEGAETH"] = 4326] = "MEGAETH";
+    ChainId[ChainId["ARC"] = 5042] = "ARC";
 })(ChainId || (exports.ChainId = ChainId = {}));
 var RequestStatus;
 (function (RequestStatus) {
