diff --git a/dist/TransactionController.cjs b/dist/TransactionController.cjs
index 0000000000000000000000000000000000000000..1111111111111111111111111111111111111111 100644
--- a/dist/TransactionController.cjs
+++ b/dist/TransactionController.cjs
@@ -94,6 +94,12 @@ const metadata = {
         includeInDebugSnapshot: false,
         usedInUi: false,
     },
+    swapsTransactions: {
+        includeInStateLogs: true,
+        persist: true,
+        includeInDebugSnapshot: false,
+        usedInUi: true,
+    },
 };
 /**
  * Multiplier used to determine a transaction's increased gas fee during cancellation
diff --git a/dist/TransactionController.mjs b/dist/TransactionController.mjs
index 0000000000000000000000000000000000000000..1111111111111111111111111111111111111111 100644
--- a/dist/TransactionController.mjs
+++ b/dist/TransactionController.mjs
@@ -96,6 +96,12 @@ const metadata = {
         includeInDebugSnapshot: false,
         usedInUi: false,
     },
+    swapsTransactions: {
+        includeInStateLogs: true,
+        persist: true,
+        includeInDebugSnapshot: false,
+        usedInUi: true,
+    },
 };
 /**
  * Multiplier used to determine a transaction's increased gas fee during cancellation
