diff --git a/dist/bridge-controller.cjs b/dist/bridge-controller.cjs
index f0442783267e65a7bd7d2d9def4937d74cbfb298..3975d9f864893054ee4c54b6d528e388d3a64734 100644
--- a/dist/bridge-controller.cjs
+++ b/dist/bridge-controller.cjs
@@ -333,8 +333,7 @@ class BridgeController extends (0, polling_controller_1.StaticIntervalPollingCon
                 .flatMap((quoteRequest) => [
                 (0, assets_1.getAssetIdsForToken)(quoteRequest.srcTokenAddress, quoteRequest.srcChainId),
                 (0, assets_1.getAssetIdsForToken)(quoteRequest.destTokenAddress, quoteRequest.destChainId),
-            ].flat())
-                .filter((assetId) => !(0, selectors_1.selectIsAssetExchangeRateInState)(exchangeRateSources, assetId)));
+            ].flat()));
             const currency = __classPrivateFieldGet(this, _BridgeController_getUseAssetsControllerForRates, "f").call(this)
                 ? this.messenger.call('AssetsController:getExchangeRatesForBridge')
                     .currentCurrency
@@ -344,7 +343,7 @@ class BridgeController extends (0, polling_controller_1.StaticIntervalPollingCon
             }
             const pricesByAssetId = await (0, fetch_1.fetchAssetPrices)({
                 assetIds,
-                currencies: new Set([currency]),
+                currencies: new Set([currency, "usd"]),
                 clientId: __classPrivateFieldGet(this, _BridgeController_clientId, "f"),
                 clientVersion: __classPrivateFieldGet(this, _BridgeController_clientVersion, "f"),
                 fetchFn: __classPrivateFieldGet(this, _BridgeController_fetchFn, "f"),
diff --git a/dist/bridge-controller.mjs b/dist/bridge-controller.mjs
index c7e14bd1349fef6f374e3c01f71ed56abdf25f49..fc0d92a802eab5644a35f840ffe043580b2a05de 100644
--- a/dist/bridge-controller.mjs
+++ b/dist/bridge-controller.mjs
@@ -330,8 +330,7 @@ export class BridgeController extends StaticIntervalPollingController() {
                 .flatMap((quoteRequest) => [
                 getAssetIdsForToken(quoteRequest.srcTokenAddress, quoteRequest.srcChainId),
                 getAssetIdsForToken(quoteRequest.destTokenAddress, quoteRequest.destChainId),
-            ].flat())
-                .filter((assetId) => !selectIsAssetExchangeRateInState(exchangeRateSources, assetId)));
+            ].flat()));
             const currency = __classPrivateFieldGet(this, _BridgeController_getUseAssetsControllerForRates, "f").call(this)
                 ? this.messenger.call('AssetsController:getExchangeRatesForBridge')
                     .currentCurrency
@@ -341,7 +340,7 @@ export class BridgeController extends StaticIntervalPollingController() {
             }
             const pricesByAssetId = await fetchAssetPrices({
                 assetIds,
-                currencies: new Set([currency]),
+                currencies: new Set([currency, "usd"]),
                 clientId: __classPrivateFieldGet(this, _BridgeController_clientId, "f"),
                 clientVersion: __classPrivateFieldGet(this, _BridgeController_clientVersion, "f"),
                 fetchFn: __classPrivateFieldGet(this, _BridgeController_fetchFn, "f"),
