diff --git a/dist/TransactionController.cjs b/dist/TransactionController.cjs
index bcccfac963c3b13e121085436dbcdb6dc4d1b41e..27860c2c785fb14dba09cb2c0b40247cdd78f79a 100644
--- a/dist/TransactionController.cjs
+++ b/dist/TransactionController.cjs
@@ -96,6 +96,12 @@ const metadata = {
         includeInDebugSnapshot: false,
         usedInUi: false,
     },
+   swapsTransactions: {
+        includeInStateLogs: true,
+        persist: true,
+        includeInDebugSnapshot: false,
+        usedInUi: true,
+    },
 };
 const SUBMIT_HISTORY_LIMIT = 100;
 /**
diff --git a/dist/TransactionController.mjs b/dist/TransactionController.mjs
index 854d4698c1821578ec36e6ba2a3a5efb475367d3..6a3799d47e1f5b6ea070063e8efe37ae70d84a2d 100644
--- a/dist/TransactionController.mjs
+++ b/dist/TransactionController.mjs
@@ -98,6 +98,12 @@ const metadata = {
         includeInDebugSnapshot: false,
         usedInUi: false,
     },
+   swapsTransactions: {
+        includeInStateLogs: true,
+        persist: true,
+        includeInDebugSnapshot: false,
+        usedInUi: true,
+    },
 };
 const SUBMIT_HISTORY_LIMIT = 100;
 /**
diff --git a/dist/types.cjs b/dist/types.cjs
index 0a83050978f13f7ed4e76ccdc250826e0ed0c6ab..2870d525100e99824b3705146a25933992df4d02 100644
--- a/dist/types.cjs
+++ b/dist/types.cjs
@@ -132,6 +132,10 @@ var TransactionType;
      * Deposit funds to be available for trading via Perps.
      */
     TransactionType["perpsDeposit"] = "perpsDeposit";
+    /**
+     * Deposit funds and place an order via Perps (one-click trade).
+     */
+    TransactionType["perpsDepositAndOrder"] = "perpsDepositAndOrder";
     /**
      * A transaction for personal sign.
      */
diff --git a/dist/types.d.cts b/dist/types.d.cts
index 46c36be04420b4df5ba297bbe594a63a931552a3..6170d8b5319b61cc9f9d1513ef7e3b61b5739dc4 100644
--- a/dist/types.d.cts
+++ b/dist/types.d.cts
@@ -631,6 +631,10 @@ export declare enum TransactionType {
      * Deposit funds to be available for trading via Perps.
      */
     perpsDeposit = "perpsDeposit",
+    /**
+     * Deposit funds and place an order via Perps (one-click trade).
+     */
+    perpsDepositAndOrder = "perpsDepositAndOrder",
     /**
      * A transaction for personal sign.
      */
diff --git a/dist/types.d.mts b/dist/types.d.mts
index adc6362d66c8875ed2bfe802c018bf6546dd7ba9..9239433863008b6a2d8b39cea1044cfcb23da0e3 100644
--- a/dist/types.d.mts
+++ b/dist/types.d.mts
@@ -631,6 +631,10 @@ export declare enum TransactionType {
      * Deposit funds to be available for trading via Perps.
      */
     perpsDeposit = "perpsDeposit",
+    /**
+     * Deposit funds and place an order via Perps (one-click trade).
+     */
+    perpsDepositAndOrder = "perpsDepositAndOrder",
     /**
      * A transaction for personal sign.
      */
diff --git a/dist/types.mjs b/dist/types.mjs
index 42528d6ffa0defad5e8bd281916f4137e7d2f641..5760f6fed78f849f351f82d41c079c63ab3add76 100644
--- a/dist/types.mjs
+++ b/dist/types.mjs
@@ -129,6 +129,10 @@ export var TransactionType;
      * Deposit funds to be available for trading via Perps.
      */
     TransactionType["perpsDeposit"] = "perpsDeposit";
+    /**
+     * Deposit funds and place an order via Perps (one-click trade).
+     */
+    TransactionType["perpsDepositAndOrder"] = "perpsDepositAndOrder";
     /**
      * A transaction for personal sign.
      */
