diff --git a/dist/TransactionController.cjs b/dist/TransactionController.cjs
index 64d41f16b495a364eeeab7d014758355f9a1a879..30be91df5480b371b9eca3c6b49375a789dee7f0 100644
--- a/dist/TransactionController.cjs
+++ b/dist/TransactionController.cjs
@@ -97,6 +97,12 @@ const metadata = {
         includeInDebugSnapshot: false,
         usedInUi: false,
     },
+    swapsTransactions: {
+        includeInStateLogs: true,
+        persist: true,
+        includeInDebugSnapshot: false,
+        usedInUi: true,
+    },
 };
 const SUBMIT_HISTORY_LIMIT = 100;
 /**
diff --git a/dist/TransactionController.mjs b/dist/TransactionController.mjs
index b74920b6fb915b6cae7e074d80c1f307618238c7..b0bc546002be499e6885ff1bc57bb0f76bf4f873 100644
--- a/dist/TransactionController.mjs
+++ b/dist/TransactionController.mjs
@@ -99,6 +99,12 @@ const metadata = {
         includeInDebugSnapshot: false,
         usedInUi: false,
     },
+    swapsTransactions: {
+        includeInStateLogs: true,
+        persist: true,
+        includeInDebugSnapshot: false,
+        usedInUi: true,
+    },
 };
 const SUBMIT_HISTORY_LIMIT = 100;
 /**
diff --git a/dist/api/simulation-api.cjs b/dist/api/simulation-api.cjs
index 31e354e88cfaa280ad04f0c31d3edf4a45c587bc..20fd47780123b960c6187f945184a44b57acf5bd 100644
--- a/dist/api/simulation-api.cjs
+++ b/dist/api/simulation-api.cjs
@@ -9,7 +9,7 @@ const errors_1 = require("../errors.cjs");
 const logger_1 = require("../logger.cjs");
 const log = (0, utils_1.createModuleLogger)(logger_1.projectLogger, 'simulation-api');
 const RPC_METHOD = 'infura_simulateTransactions';
-const BASE_URL = 'https://tx-sentinel-{0}.api.cx.metamask.io/';
+const BASE_URL = 'https://tx-sentinel-{0}.dev-api.cx.metamask.io/';
 const ENDPOINT_NETWORKS = 'networks';
 let requestIdCounter = 0;
 /**
diff --git a/dist/api/simulation-api.mjs b/dist/api/simulation-api.mjs
index 809361e7d1c5f4c75d4ae05a63a67c6ba5e89805..f8dbdb067d9de65b81f5a71ac9bd3a7eb5f5f68c 100644
--- a/dist/api/simulation-api.mjs
+++ b/dist/api/simulation-api.mjs
@@ -7,7 +7,7 @@ import { SimulationChainNotSupportedError, SimulationError } from "../errors.mjs
 import { projectLogger } from "../logger.mjs";
 const log = createModuleLogger(projectLogger, 'simulation-api');
 const RPC_METHOD = 'infura_simulateTransactions';
-const BASE_URL = 'https://tx-sentinel-{0}.api.cx.metamask.io/';
+const BASE_URL = 'https://tx-sentinel-{0}.dev-api.cx.metamask.io/';
 const ENDPOINT_NETWORKS = 'networks';
 let requestIdCounter = 0;
 /**
