diff --git a/dist/bridge-controller.cjs b/dist/bridge-controller.cjs
index a9a8d64718630515cec06d5e7ae60e96c37adf7e..137c86e4bdc01466d9d49419d2a42caee8657140 100644
--- a/dist/bridge-controller.cjs
+++ b/dist/bridge-controller.cjs
@@ -680,11 +680,11 @@ _BridgeController_abortController = new WeakMap(), _BridgeController_quotesFirst
     const networkClient = this.messagingSystem.call('NetworkController:getNetworkClientById', selectedNetworkClientId);
     return networkClient;
 }, _BridgeController_getNetworkClientByChainId = function _BridgeController_getNetworkClientByChainId(chainId) {
-    const networkClientId = this.messenger.call('NetworkController:findNetworkClientIdByChainId', chainId);
+    const networkClientId = this.messagingSystem.call('NetworkController:findNetworkClientIdByChainId', chainId);
     if (!networkClientId) {
         throw new Error(`No network client found for chainId: ${chainId}`);
     }
-    const networkClient = this.messenger.call('NetworkController:getNetworkClientById', networkClientId);
+    const networkClient = this.messagingSystem.call('NetworkController:getNetworkClientById', networkClientId);
     return networkClient;
 };
 //# sourceMappingURL=bridge-controller.cjs.map
