diff --git a/dist/bridge-status-controller.cjs b/dist/bridge-status-controller.cjs
index b787174f2c8c448ed1ad9c8884204c5c8b6858be..af058623871badb4891564c003693ea19d0aa676 100644
--- a/dist/bridge-status-controller.cjs
+++ b/dist/bridge-status-controller.cjs
@@ -834,7 +834,12 @@ class BridgeStatusController extends (0, polling_controller_1.StaticIntervalPoll
                         ? quoteResponse.approval
                         : undefined, quoteResponse.resetApproval, requireApproval);
                     approvalTxId = approvalTxMeta?.id;
-                    await (0, transaction_1.handleMobileHardwareWalletDelay)(requireApproval);
+                    if (requireApproval && approvalTxMeta) {
+                        await __classPrivateFieldGet(this, _BridgeStatusController_waitForTxConfirmation, "f").call(this, approvalTxMeta.id);
+                    }
+                    else {
+                        await (0, transaction_1.handleMobileHardwareWalletDelay)(requireApproval);
+                    }
                     // Generate actionId for pre-submission history (non-batch EVM only)
                     const actionId = (0, transaction_1.generateActionId)().toString();
                     // Add pre-submission history keyed by actionId
diff --git a/dist/bridge-status-controller.mjs b/dist/bridge-status-controller.mjs
index 2fe71bdd2caf4d62f7946e9466b31367d360cd7c..fb9c0bc45abf88873452667b85ef2ea0cdfd929c 100644
--- a/dist/bridge-status-controller.mjs
+++ b/dist/bridge-status-controller.mjs
@@ -831,7 +831,12 @@ export class BridgeStatusController extends StaticIntervalPollingController() {
                         ? quoteResponse.approval
                         : undefined, quoteResponse.resetApproval, requireApproval);
                     approvalTxId = approvalTxMeta?.id;
-                    await handleMobileHardwareWalletDelay(requireApproval);
+                    if (requireApproval && approvalTxMeta) {
+                        await __classPrivateFieldGet(this, _BridgeStatusController_waitForTxConfirmation, "f").call(this, approvalTxMeta.id);
+                    }
+                    else {
+                        await handleMobileHardwareWalletDelay(requireApproval);
+                    }
                     // Generate actionId for pre-submission history (non-batch EVM only)
                     const actionId = generateActionId().toString();
                     // Add pre-submission history keyed by actionId
