diff --git a/dist/utils/hyperLiquidAdapter.cjs b/dist/utils/hyperLiquidAdapter.cjs
index 11b8fa878e6ba3ad0633cd90e48ac2c1db8f5ff7..0d474380769af6790539ad305d026c5e475ebef7 100644
--- a/dist/utils/hyperLiquidAdapter.cjs
+++ b/dist/utils/hyperLiquidAdapter.cjs
@@ -122,11 +122,11 @@ function adaptOrderFromSDK(rawOrder, position) {
         rawOrder.children.forEach((child) => {
             if (child.isTrigger && child.orderType) {
                 if (child.orderType.includes('Take Profit')) {
-                    takeProfitPrice = child.triggerPx ?? child.limitPx;
+                    takeProfitPrice = child.triggerPx || child.limitPx;
                     takeProfitOrderId = child.oid.toString();
                 }
                 else if (child.orderType.includes('Stop')) {
-                    stopLossPrice = child.triggerPx ?? child.limitPx;
+                    stopLossPrice = child.triggerPx || child.limitPx;
                     stopLossOrderId = child.oid.toString();
                 }
             }
diff --git a/dist/utils/hyperLiquidAdapter.mjs b/dist/utils/hyperLiquidAdapter.mjs
index 9c5206ddb36fb03cb540cd7d6b36f1206ce51243..a91fce44cec3126caaac99cfa0d8d2c2484a6e6f 100644
--- a/dist/utils/hyperLiquidAdapter.mjs
+++ b/dist/utils/hyperLiquidAdapter.mjs
@@ -117,11 +117,11 @@ export function adaptOrderFromSDK(rawOrder, position) {
         rawOrder.children.forEach((child) => {
             if (child.isTrigger && child.orderType) {
                 if (child.orderType.includes('Take Profit')) {
-                    takeProfitPrice = child.triggerPx ?? child.limitPx;
+                    takeProfitPrice = child.triggerPx || child.limitPx;
                     takeProfitOrderId = child.oid.toString();
                 }
                 else if (child.orderType.includes('Stop')) {
-                    stopLossPrice = child.triggerPx ?? child.limitPx;
+                    stopLossPrice = child.triggerPx || child.limitPx;
                     stopLossOrderId = child.oid.toString();
                 }
             }
