diff --git a/dist/TransactionController.cjs b/dist/TransactionController.cjs
index 64d41f16b495a364eeeab7d014758355f9a1a879..9648bde2c018274e6ae082dafef6878a64281776 100644
--- a/dist/TransactionController.cjs
+++ b/dist/TransactionController.cjs
@@ -97,6 +97,12 @@ const metadata = {
         includeInDebugSnapshot: false,
         usedInUi: false,
     },
+   swapsTransactions: {
+        includeInStateLogs: true,
+        persist: true,
+        includeInDebugSnapshot: false,
+        usedInUi: true,
+    },
 };
 const SUBMIT_HISTORY_LIMIT = 100;
 /**
diff --git a/dist/TransactionController.mjs b/dist/TransactionController.mjs
index b74920b6fb915b6cae7e074d80c1f307618238c7..05aaef45926e813c76809b442cd27b481db2fd74 100644
--- a/dist/TransactionController.mjs
+++ b/dist/TransactionController.mjs
@@ -99,6 +99,12 @@ const metadata = {
         includeInDebugSnapshot: false,
         usedInUi: false,
     },
+   swapsTransactions: {
+        includeInStateLogs: true,
+        persist: true,
+        includeInDebugSnapshot: false,
+        usedInUi: true,
+    },
 };
 const SUBMIT_HISTORY_LIMIT = 100;
 /**
