diff --git a/dist/typography.cjs b/dist/typography.cjs
index e8220d2eecd63f95c727463a01c2be679e1028df..0d7177be94410321e167b7b14a5fb59bf38ac15a 100644
--- a/dist/typography.cjs
+++ b/dist/typography.cjs
@@ -104,12 +104,12 @@ exports.typographyTailwindConfig = {
         ],
     },
     fontFamily: {
-        'default-regular': 'Geist-Regular',
-        'default-regular-italic': 'Geist-RegularItalic',
-        'default-medium': 'Geist-Medium',
-        'default-medium-italic': 'Geist-MediumItalic',
-        'default-bold': 'Geist-SemiBold',
-        'default-bold-italic': 'Geist-SemiBoldItalic',
+        'default-regular': 'Oswald-Regular',
+        'default-regular-italic': 'Oswald-Regular',
+        'default-medium': 'Oswald-Medium',
+        'default-medium-italic': 'Oswald-Medium',
+        'default-bold': 'Oswald-SemiBold',
+        'default-bold-italic': 'Oswald-SemiBold',
         'accent-regular': 'MMSans-Regular',
         'accent-medium': 'MMSans-Medium',
         'accent-bold': 'MMSans-Bold',
diff --git a/dist/typography.mjs b/dist/typography.mjs
index 908003a3a3fcee2f5b50dc496120824260bc9b5e..1dd434ed8662f5f207bdd9a5d8b3368a235b4642 100644
--- a/dist/typography.mjs
+++ b/dist/typography.mjs
@@ -101,12 +101,12 @@ export const typographyTailwindConfig = {
         ],
     },
     fontFamily: {
-        'default-regular': 'Geist-Regular',
-        'default-regular-italic': 'Geist-RegularItalic',
-        'default-medium': 'Geist-Medium',
-        'default-medium-italic': 'Geist-MediumItalic',
-        'default-bold': 'Geist-SemiBold',
-        'default-bold-italic': 'Geist-SemiBoldItalic',
+        'default-regular': 'Oswald-Regular',
+        'default-regular-italic': 'Oswald-Regular',
+        'default-medium': 'Oswald-Medium',
+        'default-medium-italic': 'Oswald-Medium',
+        'default-bold': 'Oswald-SemiBold',
+        'default-bold-italic': 'Oswald-SemiBold',
         'accent-regular': 'MMSans-Regular',
         'accent-medium': 'MMSans-Medium',
         'accent-bold': 'MMSans-Bold',
