diff --git a/dist/multi-chain-accounts-service/api-balance-fetcher.cjs b/dist/multi-chain-accounts-service/api-balance-fetcher.cjs
index 340fffe498b3fec0c0b335c1fea4b816926d945c..8bff516e428a2960d2dee85cf7e95f72f2bd01aa 100644
--- a/dist/multi-chain-accounts-service/api-balance-fetcher.cjs
+++ b/dist/multi-chain-accounts-service/api-balance-fetcher.cjs
@@ -88,6 +88,10 @@ class AccountsApiBalanceFetcher {
                 }
                 const account = checksum(addressPart);
                 const token = checksum(b.address);
+                // Use original address for zero address tokens, checksummed for others
+                // TODO: this is a hack to get the correct account address type but needs to be fixed
+                // by mgrating tokenBalancesController to checksum addresses
+                const finalAccount = token === ZERO_ADDRESS ? account : addressPart;
                 const chainId = (0, controller_utils_1.toHex)(b.chainId);
                 let value;
                 try {
@@ -108,13 +112,13 @@ class AccountsApiBalanceFetcher {
                 }
                 // Track native balances for later
                 if (token === ZERO_ADDRESS && value !== undefined) {
-                    nativeBalancesFromAPI.set(`${account}-${chainId}`, value);
+                    nativeBalancesFromAPI.set(`${finalAccount}-${chainId}`, value);
                 }
                 return [
                     {
                         success: value !== undefined,
                         value,
-                        account,
+                        account: finalAccount,
                         token,
                         chainId,
                     },
