diff --git a/dist/utils/fetch-server-events.cjs b/dist/utils/fetch-server-events.cjs
index 1d085e5d91ae56683918ab10efafb8fced96ac96..1574836822c31c0ee60b62f8a19464668ad25217 100644
--- a/dist/utils/fetch-server-events.cjs
+++ b/dist/utils/fetch-server-events.cjs
@@ -1,6 +1,7 @@
 "use strict";
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.fetchServerEvents = void 0;
+const fetch_1 = require("expo/fetch");
 /**
  * Streams server-sent events from the given URL
  *
@@ -13,7 +14,7 @@ exports.fetchServerEvents = void 0;
  */
 const fetchServerEvents = async (url, { onMessage, onError, onClose, fetchFn, ...requestOptions }) => {
     try {
-        const response = await fetchFn(url, requestOptions);
+        const response = await (0, fetch_1.fetch)(url, requestOptions);
         if (!response.ok || !response.body) {
             throw new Error(`${response.status}`);
         }
