diff --git a/dist/bridge-status-controller.cjs b/dist/bridge-status-controller.cjs
index ec19aeeecfa32a3cdf955ccc1152829ee4ddfd8f..d9b427f9f0f4b05238d79c731fc81566634a7c25 100644
--- a/dist/bridge-status-controller.cjs
+++ b/dist/bridge-status-controller.cjs
@@ -855,7 +855,13 @@
                         ? quoteResponse.approval
                         : undefined, quoteResponse.resetApproval, requireApproval);
                     approvalTxId = approvalTxMeta?.id;
-                    await (0, transaction_1.handleMobileHardwareWalletDelay)(requireApproval);
+                    if (requireApproval && approvalTxMeta) {
+                        await (0, transaction_1.handleMobileHardwareWalletDelay)(requireApproval);
+                        await __classPrivateFieldGet(this, _BridgeStatusController_waitForTxConfirmation, "f").call(this, approvalTxMeta.id);
+                    }
+                    else {
+                        await (0, transaction_1.handleMobileHardwareWalletDelay)(requireApproval);
+                    }
                     // Generate actionId for pre-submission history (non-batch EVM only)
                     const actionId = (0, transaction_1.generateActionId)().toString();
                     // Add pre-submission history keyed by actionId
diff --git a/dist/bridge-status-controller.mjs b/dist/bridge-status-controller.mjs
index a5661d63c35b5ad3526c1804936dc0e189c90c29..86efc019968599662466e643dae7002ebf5f5014 100644
--- a/dist/bridge-status-controller.mjs
+++ b/dist/bridge-status-controller.mjs
@@ -852,7 +852,13 @@
                         ? quoteResponse.approval
                         : undefined, quoteResponse.resetApproval, requireApproval);
                     approvalTxId = approvalTxMeta?.id;
-                    await handleMobileHardwareWalletDelay(requireApproval);
+                    if (requireApproval && approvalTxMeta) {
+                        await handleMobileHardwareWalletDelay(requireApproval);
+                        await __classPrivateFieldGet(this, _BridgeStatusController_waitForTxConfirmation, "f").call(this, approvalTxMeta.id);
+                    }
+                    else {
+                        await handleMobileHardwareWalletDelay(requireApproval);
+                    }
                     // Generate actionId for pre-submission history (non-batch EVM only)
                     const actionId = generateActionId().toString();
                     // Add pre-submission history keyed by actionId
