diff --git a/dist/caip-types.cjs b/dist/caip-types.cjs
index fa14cd686856ab85f76d181c7830b1ab86bd60e2..586774bb29ce50278bc3aa344d6175908f1065da 100644
--- a/dist/caip-types.cjs
+++ b/dist/caip-types.cjs
@@ -182,7 +182,7 @@ exports.isCaipAssetId = isCaipAssetId;
 function parseCaipChainId(caipChainId) {
     const match = exports.CAIP_CHAIN_ID_REGEX.exec(caipChainId);
     if (!match?.groups) {
-        throw new Error('Invalid CAIP chain ID.');
+        throw new Error('Invalid CAIP chain ID. CAIPDBG input=' + JSON.stringify(caipChainId) + ' typeof=' + typeof caipChainId + ' stack=' + new Error('x').stack);
     }
     return {
         namespace: match.groups.namespace,
diff --git a/dist/caip-types.mjs b/dist/caip-types.mjs
index 1a3e7c7926a703850164c71a192236335a28e8cf..3caf5f57ed01fd2dfc1532efd5ab491df624b603 100644
--- a/dist/caip-types.mjs
+++ b/dist/caip-types.mjs
@@ -169,7 +169,7 @@ export function isCaipAssetId(value) {
 export function parseCaipChainId(caipChainId) {
     const match = CAIP_CHAIN_ID_REGEX.exec(caipChainId);
     if (!match?.groups) {
-        throw new Error('Invalid CAIP chain ID.');
+        throw new Error('Invalid CAIP chain ID. CAIPDBG input=' + JSON.stringify(caipChainId) + ' typeof=' + typeof caipChainId + ' stack=' + new Error('x').stack);
     }
     return {
         namespace: match.groups.namespace,
