diff --git a/lib/esm/fetch.js b/lib/esm/fetch.js
index 31eb2278da53ba79d9fc78ea32d69f2d15f325ff..a464e27484fe775ad92d00c26c4c09c1d6e55600 100644
--- a/lib/esm/fetch.js
+++ b/lib/esm/fetch.js
@@ -37,9 +37,9 @@ export function fetchEventSource(input, _a) {
             window.clearTimeout(retryTimer);
             curRequestController.abort();
         }
-        inputSignal === null || inputSignal === void 0 ? void 0 : inputSignal.addEventListener('abort', () => {
+        inputSignal === null || inputSignal === void 0 ? void 0 : inputSignal.addEventListener('abort', (e) => {
             dispose();
-            resolve();
+            reject(Object.assign(new Error('Aborted'), { name: 'AbortError', message: e.target.reason }));
         });
         const fetch = inputFetch !== null && inputFetch !== void 0 ? inputFetch : window.fetch;
         const onopen = inputOnOpen !== null && inputOnOpen !== void 0 ? inputOnOpen : defaultOnOpen;
