diff --git a/android/build.gradle b/android/build.gradle
index 13c7e43faf79865f9c9aef8d4d4971edf5057449..14ab1bef34ea1b93f377aa72cf086da4b32db905 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -31,7 +31,7 @@ buildscript {
     if (project == rootProject) {
         repositories {
             google()
-            jcenter()
+            mavenCentral()
         }
         dependencies {
             classpath 'com.android.tools.build:gradle:3.5.3'
@@ -67,7 +67,7 @@ repositories {
         url "$rootDir/../node_modules/jsc-android/dist"
     }
     google()
-    jcenter()
+    mavenCentral()
 }
 
 dependencies {
