diff --git a/dist/providers/browserstack/index.js b/dist/providers/browserstack/index.js
index c84d31cb938f284c04051bcca7245ccbe42fa684..58a8bc4f521fd46a34d6caba7b0a88170e7ac6a1 100644
--- a/dist/providers/browserstack/index.js
+++ b/dist/providers/browserstack/index.js
@@ -248,7 +248,7 @@ class BrowserStackDeviceProvider {
             capabilities: {
                 "bstack:options": {
                     debug: true,
-                    local: true,
+                    local: process.env.BROWSERSTACK_LOCAL,
                     ...(process.env.BROWSERSTACK_LOCAL_IDENTIFIER ? { localIdentifier: process.env.BROWSERSTACK_LOCAL_IDENTIFIER } : {}),
                     networkProfile : '4g-lte-advanced-good',
                     interactiveDebugging: true,
