diff --git a/dist/providers/browserstack/index.js b/dist/providers/browserstack/index.js
index 61ecd1508400f59302d6139a805e7a912e9a1036..c7772bec15413478dd0d446f7c9f805e16b00851 100644
--- a/dist/providers/browserstack/index.js
+++ b/dist/providers/browserstack/index.js
@@ -104,6 +104,21 @@ class BrowserStackDeviceProvider {
     async createDriver(config) {
         const WebDriver = (await import("webdriver")).default;
         const webDriverClient = await WebDriver.newSession(config);
+        await webDriverClient.updateSettings({
+            waitForIdleTimeout: 100,
+            waitForSelectorTimeout: 0,
+            shouldWaitForQuiescence: false,
+            snapshotMaxDepth: 62,
+            snapshotTimeout: 50000,
+            "appium:settings[snapshotMaxDepth]": 62,
+            "appium:settings[customSnapshotTimeout]": 50000,
+            "appium:bstackPageSource": {
+                "enable": true,
+                "samplesX": 15,
+                "samplesY": 15,
+                "maxDepth": 100
+            }
+        });
         this.sessionId = webDriverClient.sessionId;
         const bundleId = await this.getAppBundleIdFromSession();
         const testOptions = {
