diff --git a/dist/providers/browserstack/index.js b/dist/providers/browserstack/index.js
--- a/dist/providers/browserstack/index.js
+++ b/dist/providers/browserstack/index.js
@@ -248,8 +248,8 @@ class BrowserStackDeviceProvider {
             capabilities: {
                 "bstack:options": {
                     debug: true,
-                    local: true,
+                    local: process.env.BROWSERSTACK_LOCAL?.toLowerCase() !== 'false',
                     ...(process.env.BROWSERSTACK_LOCAL_IDENTIFIER ? { localIdentifier: process.env.BROWSERSTACK_LOCAL_IDENTIFIER } : {}),
                     ...(process.env.BROWSERSTACK_RN_PLAYGROUND_URL ? { otherApps: [process.env.BROWSERSTACK_RN_PLAYGROUND_URL] } : {}),
                     networkProfile : '4g-lte-advanced-good',
                     interactiveDebugging: true,
