diff --git a/android/build.gradle b/android/build.gradle
index 2e3129faf29f664cc3afa7e3adc3dee1a6ab8449..cf05ac42017642e15660f9564b467404010a00ba 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -29,6 +29,11 @@ android {
     targetSdkVersion safeExtGet('targetSdkVersion', DEFAULT_TARGET_SDK_VERSION)
     versionCode 1
     versionName '1.0'
+    externalNativeBuild {
+      cmake {
+        arguments "-DANDROID_SUPPORT_FLEXIBLE_PAGE_SIZES=ON"
+      }
+    }
   }
   lintOptions {
     abortOnError false
