diff --git a/android/build.gradle b/android/build.gradle
index 6674cc7715c39ba3c8802068a54fc217c0208141..87ca15cebb89e3ad1a941a6d5548a5832db2b93b 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -108,12 +108,12 @@ afterEvaluate { project ->
     }
 
     task androidJavadocJar(type: Jar, dependsOn: androidJavadoc) {
-        classifier = 'javadoc'
-        from androidJavadoc.destinationDir
+        archiveClassifier = 'javadoc'
+        from androidJavadoc.destinationDirectory
     }
 
     task androidSourcesJar(type: Jar) {
-        classifier = 'sources'
+        archiveClassifier = 'sources'
         from android.sourceSets.main.java.sourceFiles
     }
 
@@ -122,7 +122,7 @@ afterEvaluate { project ->
         def javaCompileTask = variant.javaCompileProvider.get()
 
         task "jar${name}"(type: Jar, dependsOn: javaCompileTask) {
-            from javaCompileTask.destinationDir
+            from javaCompileTask.destinationDirectory
         }
     }
 
