diff --git a/android/build.gradle b/android/build.gradle
index 7055b60151a2cbbf7aa1727dcb9515b641f74038..3e9eedea016598daebd354966788540cf4224561 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -4,7 +4,7 @@ def safeExtGet(prop, fallback) {

 buildscript {
     repositories {
-        jcenter()
+        mavenCentral()
     }

     dependencies {
@@ -44,5 +44,5 @@ repositories {
 }

 dependencies {
-    compile 'com.facebook.react:react-native:+'
+    implementation 'com.facebook.react:react-native:+'
 }
