diff --git a/android/src/main/java/com/releaseprofiler/ReleaseProfilerModule.kt b/android/src/main/java/com/releaseprofiler/ReleaseProfilerModule.kt
index 34f4cec7608eb12a5b78f01a40363d8da5f3f6dc..c70a498fa52e22361dfc574e1c5c5de56542a16a 100644
--- a/android/src/main/java/com/releaseprofiler/ReleaseProfilerModule.kt
+++ b/android/src/main/java/com/releaseprofiler/ReleaseProfilerModule.kt
@@ -79,6 +79,15 @@ class ReleaseProfilerModule(reactContext: ReactApplicationContext) :
     promise.resolve(outputPath)
   }
 
+  @ReactMethod
+  fun dumpProfiling(promise: Promise) {
+    val tempFile = File.createTempFile(
+      "sampling-profiler-trace", ".cpuprofile", reactContext.getCacheDir())
+    val outputPath = tempFile.getPath()
+    HermesSamplingProfiler.dumpSampledTraceToFile(outputPath)
+    promise.resolve(outputPath)
+  }
+
   companion object {
     const val NAME = "ReleaseProfiler"
   }
diff --git a/lib/commonjs/index.js b/lib/commonjs/index.js
index a79a437ab6b6454859132f654d7f354e6c820997..63509a3f11fe7320b0e93202b7e8ce9b7065e91d 100644
--- a/lib/commonjs/index.js
+++ b/lib/commonjs/index.js
@@ -4,6 +4,7 @@ Object.defineProperty(exports, "__esModule", {
   value: true
 });
 exports.startProfiling = startProfiling;
+exports.dumpProfiling = dumpProfiling;
 exports.stopProfiling = stopProfiling;
 var _reactNative = require("react-native");
 const LINKING_ERROR = `The package 'react-native-release-profiler' doesn't seem to be linked. Make sure: \n\n` + _reactNative.Platform.select({
@@ -18,6 +19,9 @@ const ReleaseProfiler = _reactNative.NativeModules.ReleaseProfiler ? _reactNativ
 function startProfiling() {
   return ReleaseProfiler.startProfiling();
 }
+function dumpProfiling() {
+  return ReleaseProfiler.dumpProfiling();
+}
 async function stopProfiling(saveToDownloads = false, _fileName = '') {
   if (saveToDownloads && _reactNative.Platform.OS === 'android' && _reactNative.Platform.Version < 29) {
     const result = await _reactNative.PermissionsAndroid.request(_reactNative.PermissionsAndroid.PERMISSIONS.WRITE_EXTERNAL_STORAGE);
diff --git a/lib/module/index.js b/lib/module/index.js
index a1964f09b2bfa075b8a6acd5319fba0d147e8eb7..bb2c864cf8def70ebe6a5104f7b8b954987bee27 100644
--- a/lib/module/index.js
+++ b/lib/module/index.js
@@ -11,6 +11,9 @@ const ReleaseProfiler = NativeModules.ReleaseProfiler ? NativeModules.ReleasePro
 export function startProfiling() {
   return ReleaseProfiler.startProfiling();
 }
+export function dumpProfiling() {
+  return ReleaseProfiler.dumpProfiling();
+}
 export async function stopProfiling(saveToDownloads = false, _fileName = '') {
   if (saveToDownloads && Platform.OS === 'android' && Platform.Version < 29) {
     const result = await PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.WRITE_EXTERNAL_STORAGE);
diff --git a/lib/typescript/src/index.d.ts b/lib/typescript/src/index.d.ts
index 2e4f5ba6180e7f7a7a6496f4feb5973a08e93a95..7babd0ab32c1636d5f55d15d81f527db31430b59 100644
--- a/lib/typescript/src/index.d.ts
+++ b/lib/typescript/src/index.d.ts
@@ -1,3 +1,4 @@
 export declare function startProfiling(): boolean;
+export declare function dumpProfiling(): Promise<string>;
 export declare function stopProfiling(saveToDownloads?: boolean, _fileName?: string): Promise<string>;
 //# sourceMappingURL=index.d.ts.map
\ No newline at end of file
diff --git a/src/index.tsx b/src/index.tsx
index 36f8fdf382a84d9baa59940c25220ef6c5a8646f..71faf87d67295c5e404967cbec812a6b9a5056e0 100644
--- a/src/index.tsx
+++ b/src/index.tsx
@@ -21,6 +21,10 @@ export function startProfiling(): boolean {
   return ReleaseProfiler.startProfiling();
 }
 
+export function dumpProfiling(): Promise<string> {
+  return ReleaseProfiler.dumpProfiling();
+}
+
 export async function stopProfiling(
   saveToDownloads = false,
   _fileName = ''
