diff --git a/node_modules/@exodus/react-native-payments/lib/js/PaymentRequest.js b/node_modules/@exodus/react-native-payments/lib/js/PaymentRequest.js
index 0ae7fcd..b9950f5 100644
--- a/node_modules/@exodus/react-native-payments/lib/js/PaymentRequest.js
+++ b/node_modules/@exodus/react-native-payments/lib/js/PaymentRequest.js
@@ -414,17 +414,7 @@ export default class PaymentRequest {
 
   _removeEventListeners() {
     // Internal Events
-    DeviceEventEmitter.removeSubscription(this._userDismissSubscription);
-    DeviceEventEmitter.removeSubscription(this._userAcceptSubscription);
-
-    if (IS_IOS) {
-      DeviceEventEmitter.removeSubscription(
-        this._shippingAddressChangeSubscription
-      );
-      DeviceEventEmitter.removeSubscription(
-        this._shippingOptionChangeSubscription
-      );
-    }
+    DeviceEventEmitter.removeAllListeners();
   }
 
   stopRequest() {
