diff --git a/node_modules/@ledgerhq/hw-app-eth/lib/modules/EIP712/index.js b/node_modules/@ledgerhq/hw-app-eth/lib/modules/EIP712/index.js
index 24251b4..7f4980f 100644
--- a/node_modules/@ledgerhq/hw-app-eth/lib/modules/EIP712/index.js
+++ b/node_modules/@ledgerhq/hw-app-eth/lib/modules/EIP712/index.js
@@ -15,7 +15,7 @@ Object.defineProperty(exports, "__esModule", { value: true });
 exports.signEIP712HashedMessage = exports.signEIP712Message = void 0;
 /* eslint-disable @typescript-eslint/no-duplicate-enum-values */
 const semver_1 = __importDefault(require("semver"));
-const index_1 = require("@ledgerhq/evm-tools/message/EIP712/index");
+const index_1 = require("@ledgerhq/evm-tools/lib/message/EIP712/index");
 const erc20_1 = require("../../services/ledger/erc20");
 const utils_1 = require("../../utils");
 const loadConfig_1 = require("../../services/ledger/loadConfig");
diff --git a/node_modules/@ledgerhq/hw-app-eth/lib/modules/EIP712/utils.js b/node_modules/@ledgerhq/hw-app-eth/lib/modules/EIP712/utils.js
index 4f09ea8..1cc7de0 100644
--- a/node_modules/@ledgerhq/hw-app-eth/lib/modules/EIP712/utils.js
+++ b/node_modules/@ledgerhq/hw-app-eth/lib/modules/EIP712/utils.js
@@ -14,7 +14,7 @@ var __importDefault = (this && this.__importDefault) || function (mod) {
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.getPayloadForFilterV2 = exports.getFilterDisplayNameAndSigBuffers = exports.getAppAndVersion = exports.getCoinRefTokensMap = exports.makeTypeEntryStructBuffer = exports.constructTypeDescByteString = exports.destructTypeFromString = exports.EIP712_TYPE_ENCODERS = exports.EIP712_TYPE_PROPERTIES = void 0;
 const bignumber_js_1 = __importDefault(require("bignumber.js"));
-const index_1 = require("@ledgerhq/evm-tools/message/index");
+const index_1 = require("@ledgerhq/evm-tools/lib/message/index");
 const utils_1 = require("../../utils");
 /**
  * @ignore for the README
diff --git a/node_modules/@ledgerhq/hw-app-eth/lib/services/ledger/erc20.js b/node_modules/@ledgerhq/hw-app-eth/lib/services/ledger/erc20.js
index 8fdedf8..c10cea2 100644
--- a/node_modules/@ledgerhq/hw-app-eth/lib/services/ledger/erc20.js
+++ b/node_modules/@ledgerhq/hw-app-eth/lib/services/ledger/erc20.js
@@ -15,7 +15,7 @@ Object.defineProperty(exports, "__esModule", { value: true });
 exports.byContractAddressAndChainId = exports.findERC20SignaturesInfo = void 0;
 const axios_1 = __importDefault(require("axios"));
 const logs_1 = require("@ledgerhq/logs");
-const index_1 = require("@ledgerhq/cryptoassets-evm-signatures/data/evm/index");
+const index_1 = require("@ledgerhq/cryptoassets-evm-signatures/lib/data/evm/index");
 const loadConfig_1 = require("./loadConfig");
 const asContractAddress = (addr) => {
     const a = addr.toLowerCase();
diff --git a/node_modules/@ledgerhq/hw-app-eth/lib/services/ledger/index.js b/node_modules/@ledgerhq/hw-app-eth/lib/services/ledger/index.js
index 86db997..29e1fe5 100644
--- a/node_modules/@ledgerhq/hw-app-eth/lib/services/ledger/index.js
+++ b/node_modules/@ledgerhq/hw-app-eth/lib/services/ledger/index.js
@@ -12,7 +12,7 @@ Object.defineProperty(exports, "__esModule", { value: true });
 const transactions_1 = require("@ethersproject/transactions");
 const abi_1 = require("@ethersproject/abi");
 const logs_1 = require("@ledgerhq/logs");
-const index_1 = require("@ledgerhq/domain-service/signers/index");
+const index_1 = require("@ledgerhq/domain-service/lib/signers/index");
 const constants_1 = require("../../modules/Uniswap/constants");
 const erc20_1 = require("./erc20");
 const Uniswap_1 = require("../../modules/Uniswap");
diff --git a/node_modules/@ledgerhq/hw-app-eth/lib/utils.js b/node_modules/@ledgerhq/hw-app-eth/lib/utils.js
index 6c5f584..f960780 100644
--- a/node_modules/@ledgerhq/hw-app-eth/lib/utils.js
+++ b/node_modules/@ledgerhq/hw-app-eth/lib/utils.js
@@ -26,7 +26,7 @@ Object.defineProperty(exports, "__esModule", { value: true });
 exports.safeChunkTransaction = exports.getV = exports.getChainIdAsUint32 = exports.getParity = exports.mergeResolutions = exports.nftSelectors = exports.tokenSelectors = exports.intAsHexBytes = exports.maybeHexBuffer = exports.hexBuffer = exports.splitPath = exports.padHexString = exports.ERC1155_CLEAR_SIGNED_SELECTORS = exports.ERC721_CLEAR_SIGNED_SELECTORS = exports.ERC20_CLEAR_SIGNED_SELECTORS = void 0;
 const bignumber_js_1 = require("bignumber.js");
 const rlp = __importStar(require("@ethersproject/rlp"));
-const index_1 = require("@ledgerhq/evm-tools/selectors/index");
+const index_1 = require("@ledgerhq/evm-tools/lib/selectors/index");
 Object.defineProperty(exports, "ERC20_CLEAR_SIGNED_SELECTORS", { enumerable: true, get: function () { return index_1.ERC20_CLEAR_SIGNED_SELECTORS; } });
 Object.defineProperty(exports, "ERC721_CLEAR_SIGNED_SELECTORS", { enumerable: true, get: function () { return index_1.ERC721_CLEAR_SIGNED_SELECTORS; } });
 Object.defineProperty(exports, "ERC1155_CLEAR_SIGNED_SELECTORS", { enumerable: true, get: function () { return index_1.ERC1155_CLEAR_SIGNED_SELECTORS; } });
