diff --git a/node_modules/@ledgerhq/react-native-hw-transport-ble/lib-es/BleTransport.js b/node_modules/@ledgerhq/react-native-hw-transport-ble/lib-es/BleTransport.js
index 3c2a74d..8fadc07 100644
--- a/node_modules/@ledgerhq/react-native-hw-transport-ble/lib-es/BleTransport.js
+++ b/node_modules/@ledgerhq/react-native-hw-transport-ble/lib-es/BleTransport.js
@@ -80,8 +80,8 @@ var __values = (this && this.__values) || function(o) {
 import Transport from "@ledgerhq/hw-transport";
 import { BleManager, ConnectionPriority, BleErrorCode, } from "react-native-ble-plx";
 import { getBluetoothServiceUuids, getInfosForServiceUuid, } from "@ledgerhq/devices";
-import { sendAPDU } from "@ledgerhq/devices/ble/sendAPDU";
-import { receiveAPDU } from "@ledgerhq/devices/ble/receiveAPDU";
+import { sendAPDU } from "@ledgerhq/devices/ble/src/sendAPDU";
+import { receiveAPDU } from "@ledgerhq/devices/ble/src/receiveAPDU";
 import { log } from "@ledgerhq/logs";
 import { defer, merge, from, of, throwError } from "rxjs";
 import { share, ignoreElements, first, map, tap, catchError, } from "rxjs/operators";
diff --git a/node_modules/@ledgerhq/react-native-hw-transport-ble/lib/BleTransport.js b/node_modules/@ledgerhq/react-native-hw-transport-ble/lib/BleTransport.js
index 3f7242a..22b0920 100644
--- a/node_modules/@ledgerhq/react-native-hw-transport-ble/lib/BleTransport.js
+++ b/node_modules/@ledgerhq/react-native-hw-transport-ble/lib/BleTransport.js
@@ -86,8 +86,8 @@ exports.setReconnectionConfig = void 0;
 var hw_transport_1 = __importDefault(require("@ledgerhq/hw-transport"));
 var react_native_ble_plx_1 = require("react-native-ble-plx");
 var devices_1 = require("@ledgerhq/devices");
-var sendAPDU_1 = require("@ledgerhq/devices/ble/sendAPDU");
-var receiveAPDU_1 = require("@ledgerhq/devices/ble/receiveAPDU");
+var sendAPDU_1 = require("@ledgerhq/devices/src/ble/sendAPDU");
+var receiveAPDU_1 = require("@ledgerhq/devices/src/ble/receiveAPDU");
 var logs_1 = require("@ledgerhq/logs");
 var rxjs_1 = require("rxjs");
 var operators_1 = require("rxjs/operators");
diff --git a/node_modules/@ledgerhq/react-native-hw-transport-ble/src/BleTransport.ts b/node_modules/@ledgerhq/react-native-hw-transport-ble/src/BleTransport.ts
index fa46d79..a4eb245 100644
--- a/node_modules/@ledgerhq/react-native-hw-transport-ble/src/BleTransport.ts
+++ b/node_modules/@ledgerhq/react-native-hw-transport-ble/src/BleTransport.ts
@@ -16,8 +16,8 @@ import {
   getInfosForServiceUuid,
 } from "@ledgerhq/devices";
 import type { DeviceModel } from "@ledgerhq/devices";
-import { sendAPDU } from "@ledgerhq/devices/ble/sendAPDU";
-import { receiveAPDU } from "@ledgerhq/devices/ble/receiveAPDU";
+import { sendAPDU } from "@ledgerhq/devices/src/ble/sendAPDU";
+import { receiveAPDU } from "@ledgerhq/devices/src/ble/receiveAPDU";
 import { log } from "@ledgerhq/logs";
 import { Observable, defer, merge, from, of, throwError } from "rxjs";
 import {
