diff --git a/node_modules/@metamask/eth-ledger-bridge-keyring/dist/ledger-transport-middleware.js b/node_modules/@metamask/eth-ledger-bridge-keyring/dist/ledger-transport-middleware.js
index b649ff2..d32b858 100644
--- a/node_modules/@metamask/eth-ledger-bridge-keyring/dist/ledger-transport-middleware.js
+++ b/node_modules/@metamask/eth-ledger-bridge-keyring/dist/ledger-transport-middleware.js
@@ -39,6 +39,12 @@ class LedgerTransportMiddleware {
      */
     setTransport(transport) {
         __classPrivateFieldSet(this, _LedgerTransportMiddleware_transport, transport, "f");
+      /**
+       * PATCH to fix the issue with iOS
+       * As the transport is set, we need to recreate the eth app instance otherwise iOS will throw disconnect error when we reconnect the device.
+       * we should remove this PATCH once the new release of library is available.
+       */
+      __classPrivateFieldSet(this, _LedgerTransportMiddleware_app, new ledger_hw_app_1.MetaMaskLedgerHwAppEth(this.getTransport()), "f");
     }
     /**
      * Method to retrieve the transport object.
@@ -58,9 +64,11 @@ class LedgerTransportMiddleware {
      * @returns An generic interface for communicating with a Ledger hardware wallet to perform operation.
      */
     getEthApp() {
-        if (!__classPrivateFieldGet(this, _LedgerTransportMiddleware_app, "f")) {
-            __classPrivateFieldSet(this, _LedgerTransportMiddleware_app, new ledger_hw_app_1.MetaMaskLedgerHwAppEth(this.getTransport()), "f");
-        }
+      /**
+       * PATCH to fix the issue with iOS
+       * Move the recreate of ethApp code to setTransport method.
+       * we should remove this PATCH once the new release of library is available.
+       */
         return __classPrivateFieldGet(this, _LedgerTransportMiddleware_app, "f");
     }
 }
