diff --git a/node_modules/@metamask/network-controller/dist/NetworkController.js b/node_modules/@metamask/network-controller/dist/NetworkController.js
index 8a57064..f8d132f 100644
--- a/node_modules/@metamask/network-controller/dist/NetworkController.js
+++ b/node_modules/@metamask/network-controller/dist/NetworkController.js
@@ -453,6 +453,8 @@ _NetworkController_ethQuery = new WeakMap(), _NetworkController_infuraProjectId
         case controller_utils_1.NetworkType.mainnet:
         case controller_utils_1.NetworkType.goerli:
         case controller_utils_1.NetworkType.sepolia:
+        case controller_utils_1.NetworkType["linea-goerli"]:
+        case controller_utils_1.NetworkType["linea-mainnet"]:
             __classPrivateFieldGet(this, _NetworkController_instances, "m", _NetworkController_setupInfuraProvider).call(this, type);
             break;
         case controller_utils_1.NetworkType.rpc:
