diff --git a/node_modules/@metamask/network-controller/dist/NetworkController.js b/node_modules/@metamask/network-controller/dist/NetworkController.js
index 80160b7..0b78ee9 100644
--- a/node_modules/@metamask/network-controller/dist/NetworkController.js
+++ b/node_modules/@metamask/network-controller/dist/NetworkController.js
@@ -104,6 +104,8 @@ class NetworkController extends base_controller_1.BaseControllerV2 {
             case controller_utils_1.NetworkType.mainnet:
             case controller_utils_1.NetworkType.goerli:
             case controller_utils_1.NetworkType.sepolia:
+            case controller_utils_1.NetworkType["linea-goerli"]:
+            case controller_utils_1.NetworkType["linea-mainnet"]:
                 this.setupInfuraProvider(type);
                 break;
             case controller_utils_1.NetworkType.localhost:
@@ -159,6 +161,8 @@ class NetworkController extends base_controller_1.BaseControllerV2 {
         return (chainId !== controller_utils_1.NetworksChainId.mainnet &&
             chainId !== controller_utils_1.NetworksChainId.goerli &&
             chainId !== controller_utils_1.NetworksChainId.sepolia &&
+            chainId !== controller_utils_1.NetworksChainId["linea-goerli"] &&
+            chainId !== controller_utils_1.NetworksChainId["linea-mainnet"] &&
             chainId !== controller_utils_1.NetworksChainId.localhost);
     }
     setupStandardProvider(rpcTarget, chainId, ticker, nickname) {
