diff --git a/node_modules/@metamask/transaction-controller/dist/TransactionController.cjs b/node_modules/@metamask/transaction-controller/dist/TransactionController.cjs
index 9bc3d74..af0c49c 100644
--- a/node_modules/@metamask/transaction-controller/dist/TransactionController.cjs
+++ b/node_modules/@metamask/transaction-controller/dist/TransactionController.cjs
@@ -87,6 +87,10 @@ const metadata = {
         persist: true,
         anonymous: false,
     },
+    swapsTransactions: {
+        persist: true,
+        anonymous: false,
+    },
 };
 const SUBMIT_HISTORY_LIMIT = 100;
 /**