diff --git a/node_modules/@metamask/bridge-controller/dist/index.mjs b/node_modules/@metamask/bridge-controller/dist/index.mjs
index 18cae1b80a..16f81c20a3 100644
--- a/node_modules/@metamask/bridge-controller/dist/index.mjs
+++ b/node_modules/@metamask/bridge-controller/dist/index.mjs
@@ -24,4 +24,6 @@ export { DEFAULT_FEATURE_FLAG_CONFIG } from "./constants/bridge.mjs";
 export { getBridgeFeatureFlags } from "./utils/feature-flags.mjs";
 export { BRIDGE_DEFAULT_SLIPPAGE } from "./utils/slippage.mjs";
 export { isValidSwapsContractAddress, getSwapsContractAddress, fetchTokens } from "./utils/swaps.mjs";
+export { fetchBridgeQuoteStream } from "./utils/fetch.mjs";
+export { appendFeesToQuotes } from "./utils/quote-fees.mjs";
 //# sourceMappingURL=index.mjs.map
\ No newline at end of file
diff --git a/node_modules/@metamask/bridge-controller/dist/index.cjs b/node_modules/@metamask/bridge-controller/dist/index.cjs
index b226af2984..441c26c2da 100644
--- a/node_modules/@metamask/bridge-controller/dist/index.cjs
+++ b/node_modules/@metamask/bridge-controller/dist/index.cjs
@@ -113,4 +113,8 @@ var swaps_2 = require("./utils/swaps.cjs");
 Object.defineProperty(exports, "isValidSwapsContractAddress", { enumerable: true, get: function () { return swaps_2.isValidSwapsContractAddress; } });
 Object.defineProperty(exports, "getSwapsContractAddress", { enumerable: true, get: function () { return swaps_2.getSwapsContractAddress; } });
 Object.defineProperty(exports, "fetchTokens", { enumerable: true, get: function () { return swaps_2.fetchTokens; } });
+var fetch_2 = require("./utils/fetch.cjs");
+Object.defineProperty(exports, "fetchBridgeQuoteStream", { enumerable: true, get: function () { return fetch_2.fetchBridgeQuoteStream; } });
+var quote_fees_1 = require("./utils/quote-fees.cjs");
+Object.defineProperty(exports, "appendFeesToQuotes", { enumerable: true, get: function () { return quote_fees_1.appendFeesToQuotes; } });
 //# sourceMappingURL=index.cjs.map
\ No newline at end of file
diff --git a/node_modules/@metamask/bridge-controller/dist/index.d.mts b/node_modules/@metamask/bridge-controller/dist/index.d.mts
index 2477f3acb3..6aeda5d02d 100644
--- a/node_modules/@metamask/bridge-controller/dist/index.d.mts
+++ b/node_modules/@metamask/bridge-controller/dist/index.d.mts
@@ -32,4 +32,6 @@ export { DEFAULT_FEATURE_FLAG_CONFIG } from "./constants/bridge.mjs";
 export { getBridgeFeatureFlags } from "./utils/feature-flags.mjs";
 export { BRIDGE_DEFAULT_SLIPPAGE } from "./utils/slippage.mjs";
 export { isValidSwapsContractAddress, getSwapsContractAddress, fetchTokens, type SwapsToken, } from "./utils/swaps.mjs";
+export { fetchBridgeQuoteStream } from "./utils/fetch.mjs";
+export { appendFeesToQuotes } from "./utils/quote-fees.mjs";
 //# sourceMappingURL=index.d.mts.map
\ No newline at end of file
diff --git a/node_modules/@metamask/bridge-controller/dist/index.d.cts b/node_modules/@metamask/bridge-controller/dist/index.d.cts
index de9f050eec..2baa77e8d0 100644
--- a/node_modules/@metamask/bridge-controller/dist/index.d.cts
+++ b/node_modules/@metamask/bridge-controller/dist/index.d.cts
@@ -32,4 +32,6 @@ export { DEFAULT_FEATURE_FLAG_CONFIG } from "./constants/bridge.cjs";
 export { getBridgeFeatureFlags } from "./utils/feature-flags.cjs";
 export { BRIDGE_DEFAULT_SLIPPAGE } from "./utils/slippage.cjs";
 export { isValidSwapsContractAddress, getSwapsContractAddress, fetchTokens, type SwapsToken, } from "./utils/swaps.cjs";
+export { fetchBridgeQuoteStream } from "./utils/fetch.cjs";
+export { appendFeesToQuotes } from "./utils/quote-fees.cjs";
 //# sourceMappingURL=index.d.cts.map
\ No newline at end of file
