diff --git a/node_modules/@metamask/rpc-errors/dist/chunk-PV5NRPSS.js b/node_modules/@metamask/rpc-errors/dist/chunk-PV5NRPSS.js
index b92b1bc..467d025 100644
--- a/node_modules/@metamask/rpc-errors/dist/chunk-PV5NRPSS.js
+++ b/node_modules/@metamask/rpc-errors/dist/chunk-PV5NRPSS.js
@@ -46,9 +46,14 @@ var JsonRpcError = class extends Error {
         serialized.data.cause = _chunkXOYARAPPjs.serializeCause.call(void 0, this.data.cause);
       }
     }
+    /**
+     * passing the stack creates a loop, possible conflicting with hermes, since it's hermes that throws an error
+     */
+    /* 
     if (this.stack) {
       serialized.stack = this.stack;
     } 
+    */
     return serialized;
   }
   /**
