diff --git a/node_modules/@metamask/keyring-controller/dist/KeyringController.cjs b/node_modules/@metamask/keyring-controller/dist/KeyringController.cjs
index 8529ba1..77871be 100644
--- a/node_modules/@metamask/keyring-controller/dist/KeyringController.cjs
+++ b/node_modules/@metamask/keyring-controller/dist/KeyringController.cjs
@@ -1379,7 +1379,8 @@ async function _KeyringController_newKeyring(type, data) {
         if (!keyring.generateRandomMnemonic) {
             throw new Error(constants_1.KeyringControllerError.UnsupportedGenerateRandomMnemonic);
         }
-        keyring.generateRandomMnemonic();
+        // This patch can be removed once all the keyrings types are updated across the monorepo, the work is in progress atm
+        await keyring.generateRandomMnemonic();
         await keyring.addAccounts(1);
     }
     await __classPrivateFieldGet(this, _KeyringController_instances, "m", _KeyringController_checkForDuplicate).call(this, type, await keyring.getAccounts());
