diff --git a/node_modules/@metamask/base-controller/dist/RestrictedControllerMessenger.d.ts b/node_modules/@metamask/base-controller/dist/RestrictedControllerMessenger.d.ts
index 423c1a9..5c62cb3 100644
--- a/node_modules/@metamask/base-controller/dist/RestrictedControllerMessenger.d.ts
+++ b/node_modules/@metamask/base-controller/dist/RestrictedControllerMessenger.d.ts
@@ -15,7 +15,7 @@ import type { ActionConstraint, ActionHandler, ControllerMessenger, EventConstra
  * @template AllowedEvent - A type union of the 'type' string for any allowed events.
  * This must not include internal events that are in the messenger's namespace.
  */
-export declare class RestrictedControllerMessenger<Namespace extends string, Action extends ActionConstraint, Event extends EventConstraint, AllowedAction extends string, AllowedEvent extends string> {
+export declare class RestrictedControllerMessenger<Namespace extends string, Action extends ActionConstraint, Event extends EventConstraint, AllowedAction extends string = string, AllowedEvent extends string = string > {
     #private;
     /**
      * Constructs a restricted controller messenger
