diff --git a/node_modules/@metamask/assets-controllers/dist/TokenListController.js b/node_modules/@metamask/assets-controllers/dist/TokenListController.js
index f783f7b..7f0343e 100644
--- a/node_modules/@metamask/assets-controllers/dist/TokenListController.js
+++ b/node_modules/@metamask/assets-controllers/dist/TokenListController.js
@@ -51,7 +51,7 @@ class TokenListController extends base_controller_1.BaseControllerV2 {
      * @param options.state - Initial state to set on this controller.
      * @param options.preventPollingOnNetworkRestart - Determines whether to prevent poilling on network restart in extension.
      */
-    constructor({ chainId, preventPollingOnNetworkRestart = false, onNetworkStateChange, interval = DEFAULT_INTERVAL, cacheRefreshThreshold = DEFAULT_THRESHOLD, messenger, state, }) {
+    constructor({ chainId,selectedAddress, preventPollingOnNetworkRestart = false, onNetworkStateChange, onPreferencesStateChange,interval = DEFAULT_INTERVAL, cacheRefreshThreshold = DEFAULT_THRESHOLD, messenger, state, }) {
         super({
             name,
             metadata,
@@ -65,6 +65,15 @@ class TokenListController extends base_controller_1.BaseControllerV2 {
         this.chainId = chainId;
         this.updatePreventPollingOnNetworkRestart(preventPollingOnNetworkRestart);
         this.abortController = new abort_controller_1.AbortController();
+        if(onPreferencesStateChange){
+            onPreferencesStateChange(({ selectedAddress: newSelectedAddress }) => {
+                const  currentSelectedAddress  = selectedAddress;
+                const isSelectedAddressChanged = newSelectedAddress !== currentSelectedAddress;
+                if (isSelectedAddressChanged ) {
+                    this.fetchTokenList();
+                }
+            });
+        }
         if (onNetworkStateChange) {
             onNetworkStateChange((networkStateOrProviderConfig) => __awaiter(this, void 0, void 0, function* () {
                 // this check for "provider" is for testing purposes, since in the extension this callback will receive
diff --git a/node_modules/@metamask/assets-controllers/dist/TokensController.js b/node_modules/@metamask/assets-controllers/dist/TokensController.js
index 4a18560..4297ef3 100644
--- a/node_modules/@metamask/assets-controllers/dist/TokensController.js
+++ b/node_modules/@metamask/assets-controllers/dist/TokensController.js
@@ -528,7 +528,7 @@ class TokensController extends base_controller_1.BaseController {
             newAllIgnoredTokens = Object.assign(Object.assign({}, allIgnoredTokens), { [chainIdToAddTokens]: newIgnoredNetworkTokens });
         }
         let newAllDetectedTokens = allDetectedTokens;
-        if (newDetectedTokens === null || newDetectedTokens === void 0 ? void 0 : newDetectedTokens.length) {
+        if (newDetectedTokens === null || newDetectedTokens === void 0 ? void 0 : !!newDetectedTokens) {
             const networkDetectedTokens = allDetectedTokens[chainIdToAddTokens];
             const newDetectedNetworkTokens = Object.assign(Object.assign({}, networkDetectedTokens), { [userAddressToAddTokens]: newDetectedTokens });
             newAllDetectedTokens = Object.assign(Object.assign({}, allDetectedTokens), { [chainIdToAddTokens]: newDetectedNetworkTokens });
