diff --git a/node_modules/@metamask/design-tokens/dist/colors/index.js b/node_modules/@metamask/design-tokens/dist/colors/index.js
index 3cbf2e9..751ef0a 100644
--- a/node_modules/@metamask/design-tokens/dist/colors/index.js
+++ b/node_modules/@metamask/design-tokens/dist/colors/index.js
@@ -98,7 +98,7 @@ const colors = {
             disabled: '#1098FC80',
         },
         secondary: {
-            default: '##F8883',
+            default: '#F8883B',
             alternative: '#FAA66C',
             muted: '#F8883B26',
             inverse: '#FCFCFC',
