diff --git a/node_modules/@metamask/swaps-controller/dist/constants.js b/node_modules/@metamask/swaps-controller/dist/constants.js
index 4c98522..523f25a 100644
--- a/node_modules/@metamask/swaps-controller/dist/constants.js
+++ b/node_modules/@metamask/swaps-controller/dist/constants.js
@@ -129,8 +129,8 @@ exports.BSC_SWAPS_TOKEN_OBJECT = {
     decimals: 18,
 };
 exports.POLYGON_SWAPS_TOKEN_OBJECT = {
-    symbol: 'MATIC',
-    name: 'Matic',
+    symbol: 'POL',
+    name: 'Polygon',
     address: exports.NATIVE_SWAPS_TOKEN_ADDRESS,
     decimals: 18,
 };
