diff --git a/node_modules/@metamask/phishing-controller/dist/PhishingController.js b/node_modules/@metamask/phishing-controller/dist/PhishingController.js
index be85782..456935a 100644
--- a/node_modules/@metamask/phishing-controller/dist/PhishingController.js
+++ b/node_modules/@metamask/phishing-controller/dist/PhishingController.js
@@ -202,10 +202,11 @@ _PhishingController_inProgressUpdate = new WeakMap(), _PhishingController_instan
         if (metamaskConfigLegacy) {
             configs.push(metamaskConfig);
         }
+
         // Correctly shaping PhishFort config.
         const phishfortConfig = {
             allowlist: [],
-            blocklist: (phishfortHotlist || []).filter((i) => !metamaskConfig.blocklist.includes(i)),
+            blocklist: phishfortHotlist,
             fuzzylist: [],
             tolerance: 0,
             name: `PhishFort`,
