diff --git a/node_modules/@metamask/keyring-controller/dist/KeyringController.js b/node_modules/@metamask/keyring-controller/dist/KeyringController.js
index fc649ea..1767a63 100644
--- a/node_modules/@metamask/keyring-controller/dist/KeyringController.js
+++ b/node_modules/@metamask/keyring-controller/dist/KeyringController.js
@@ -576,6 +576,18 @@ class KeyringController extends base_controller_1.BaseController {
                 return { type, data };
             })));
             serializedKeyrings.push(...__classPrivateFieldGet(this, _KeyringController_unsupportedKeyrings, "f"));
+             /**
+             * ============================== PATCH INFORMATION ==============================
+             * The HD keyring is the default keyring for all wallets if this keyring is missing
+             * for some reason we should avoid saving the keyrings
+             *
+             * The upstream fix is here: https://github.com/MetaMask/core/pull/4168
+             *
+             * This patch can be found on the core branch `extension-keyring-controller-v13-patch`
+             */
+             if (!serializedKeyrings.some((keyring) => keyring.type === KeyringTypes.hd)) {
+                throw new Error(constants_1.KeyringControllerError.NoHdKeyring);
+            }
             let vault;
             let newEncryptionKey;
             if (__classPrivateFieldGet(this, _KeyringController_cacheEncryptionKey, "f")) {
@@ -1043,6 +1055,28 @@ class KeyringController extends base_controller_1.BaseController {
             }
         });
     }
+
+  /**
+   +   * This Patch is generated to support the multiple accounts support for ledger devices,
+   +   * The reason to create this is to access the #keyring private method is not possible from the outside.
+   +   * To add multiple accounts support for ledger Hardware wallet
+   +   *
+   +   * @param index - Index of the account to unlock
+   +   * @param keyring - Keyring instance to unlock
+   +   * @return {*} - Promise resolving to the current state
+   +   */
+   unlockLedgerWalletAccount(index, keyring) {
+        return __awaiter(this, void 0, void 0, function* () {
+          keyring.setAccountToUnlock(index);
+          // QRKeyring is not yet compatible with Keyring from
+          // @metamask/utils, but we can use the `addNewAccount` method
+          // as it internally calls `addAccounts` from on the keyring instance,
+          // which is supported by QRKeyring API.
+          yield this.addNewAccountForKeyring(keyring);
+          yield this.persistAllKeyrings();
+        });
+   }
+
     unlockQRHardwareWalletAccount(index) {
         return __awaiter(this, void 0, void 0, function* () {
             const keyring = yield this.getOrAddQRKeyring();
@@ -1091,10 +1125,17 @@ _KeyringController_keyringBuilders = new WeakMap(), _KeyringController_keyrings
     return __classPrivateFieldGet(this, _KeyringController_keyringBuilders, "f").find((keyringBuilder) => keyringBuilder.type === type);
 }, _KeyringController_addQRKeyring = function _KeyringController_addQRKeyring() {
     return __awaiter(this, void 0, void 0, function* () {
-        // QRKeyring is not yet compatible with Keyring type from @metamask/utils
-        const qrKeyring = (yield __classPrivateFieldGet(this, _KeyringController_instances, "m", _KeyringController_newKeyring).call(this, KeyringTypes.qr, {
-            accounts: [],
-        }));
+         // QRKeyring is not yet compatible with Keyring type from @metamask/utils
+        /**
+         * Patch for @metamask/keyring-controller v13.0.0
+         * Below code change will fix the issue 23804, The intial code added a empty accounts as argument when creating a new QR keyring.
+         * cause the new Keystone MetamaskKeyring default properties all are undefined during deserialise() process.
+         * Please refer to PR 23903 for detail.
+         *
+         * This patch can be found on the core branch `extension-keyring-controller-v13-patch`
+         */
+        // @ts-expect-error See patch note
+        const qrKeyring = (yield __classPrivateFieldGet(this, _KeyringController_instances, "m", _KeyringController_newKeyring).call(this, KeyringTypes.qr));
         const accounts = yield qrKeyring.getAccounts();
         yield __classPrivateFieldGet(this, _KeyringController_instances, "m", _KeyringController_checkForDuplicate).call(this, KeyringTypes.qr, accounts);
         __classPrivateFieldGet(this, _KeyringController_keyrings, "f").push(qrKeyring);
diff --git a/node_modules/@metamask/keyring-controller/dist/constants.d.ts b/node_modules/@metamask/keyring-controller/dist/constants.d.ts
index 0c02177..805c0d5 100644
--- a/node_modules/@metamask/keyring-controller/dist/constants.d.ts
+++ b/node_modules/@metamask/keyring-controller/dist/constants.d.ts
@@ -25,6 +25,7 @@ export declare enum KeyringControllerError {
     MissingVaultData = "KeyringController - Cannot persist vault without vault information",
     ExpiredCredentials = "KeyringController - Encryption key and salt provided are expired",
     NoKeyringBuilder = "KeyringController - No keyringBuilder found for keyring",
-    DataType = "KeyringController - Incorrect data type provided"
+    DataType = "KeyringController - Incorrect data type provided",
+    NoHdKeyring = "KeyringController - No HD Keyring found"
 }
 //# sourceMappingURL=constants.d.ts.map
\ No newline at end of file
diff --git a/node_modules/@metamask/keyring-controller/dist/constants.js b/node_modules/@metamask/keyring-controller/dist/constants.js
index 58b3a15..10768a8 100644
--- a/node_modules/@metamask/keyring-controller/dist/constants.js
+++ b/node_modules/@metamask/keyring-controller/dist/constants.js
@@ -30,5 +30,6 @@ var KeyringControllerError;
     KeyringControllerError["ExpiredCredentials"] = "KeyringController - Encryption key and salt provided are expired";
     KeyringControllerError["NoKeyringBuilder"] = "KeyringController - No keyringBuilder found for keyring";
     KeyringControllerError["DataType"] = "KeyringController - Incorrect data type provided";
+    KeyringControllerError["NoHdKeyring"] = "KeyringController - No HD Keyring found";
 })(KeyringControllerError = exports.KeyringControllerError || (exports.KeyringControllerError = {}));
 //# sourceMappingURL=constants.js.map
\ No newline at end of file
