diff --git a/node_modules/@metamask/controller-utils/dist/constants.d.ts b/node_modules/@metamask/controller-utils/dist/constants.d.ts
index dd0be20..56cfc46 100644
--- a/node_modules/@metamask/controller-utils/dist/constants.d.ts
+++ b/node_modules/@metamask/controller-utils/dist/constants.d.ts
@@ -61,6 +61,13 @@ export declare const BUILT_IN_NETWORKS: {
             readonly blockExplorerUrl: "https://explorer.goerli.linea.build";
         };
     };
+    readonly "linea-sepolia": {
+        readonly chainId: "0xe705";
+        readonly ticker: typeof NetworksTicker["linea-sepolia"];
+        readonly rpcPrefs: {
+            readonly blockExplorerUrl: "https://sepolia.lineascan.build";
+        };
+    };
     readonly "linea-mainnet": {
         readonly chainId: "0xe708";
         readonly ticker: NetworksTicker.mainnet;
@@ -74,9 +81,8 @@ export declare const BUILT_IN_NETWORKS: {
         readonly rpcPrefs: undefined;
     };
 };
-export declare const OPENSEA_PROXY_URL = "https://proxy.metafi.codefi.network/opensea/v1/api/v1";
-export declare const OPENSEA_API_URL = "https://api.opensea.io/api/v1";
-export declare const OPENSEA_TEST_API_URL = "https://testnets-api.opensea.io/api/v1";
+export declare const OPENSEA_PROXY_URL = "https://proxy.metafi.codefi.network/opensea/v1/api/v2";
+
 export declare const ORIGIN_METAMASK = "metamask";
 /**
  * Approval request types for various operations.
diff --git a/node_modules/@metamask/controller-utils/dist/constants.js b/node_modules/@metamask/controller-utils/dist/constants.js
index 90ede0b..d5f7202 100644
--- a/node_modules/@metamask/controller-utils/dist/constants.js
+++ b/node_modules/@metamask/controller-utils/dist/constants.js
@@ -38,6 +38,7 @@ exports.TESTNET_TICKER_SYMBOLS = {
     GOERLI: 'GoerliETH',
     SEPOLIA: 'SepoliaETH',
     LINEA_GOERLI: 'LineaETH',
+    LINEA_SEPOLIA: 'LineaETH',
 };
 /**
  * Map of all build-in Infura networks to their network, ticker and chain IDs.
@@ -71,6 +72,13 @@ exports.BUILT_IN_NETWORKS = {
             blockExplorerUrl: 'https://explorer.goerli.linea.build',
         },
     },
+    [types_1.NetworkType['linea-sepolia']]: {
+        chainId: types_1.ChainId['linea-sepolia'],
+        ticker: types_1.NetworksTicker['linea-sepolia'],
+        rpcPrefs: {
+            blockExplorerUrl: 'https://sepolia.lineascan.build',
+        },
+    },
     [types_1.NetworkType['linea-mainnet']]: {
         chainId: types_1.ChainId['linea-mainnet'],
         ticker: types_1.NetworksTicker['linea-mainnet'],
@@ -85,9 +93,7 @@ exports.BUILT_IN_NETWORKS = {
     },
 };
 // APIs
-exports.OPENSEA_PROXY_URL = 'https://proxy.metafi.codefi.network/opensea/v1/api/v1';
-exports.OPENSEA_API_URL = 'https://api.opensea.io/api/v1';
-exports.OPENSEA_TEST_API_URL = 'https://testnets-api.opensea.io/api/v1';
+exports.OPENSEA_PROXY_URL = 'https://proxy.metafi.codefi.network/opensea/v1/api/v2';
 // Default origin for controllers
 exports.ORIGIN_METAMASK = 'metamask';
 /**
@@ -121,6 +127,7 @@ exports.NETWORK_ID_TO_ETHERS_NETWORK_NAME_MAP = {
     [types_1.NetworkId.sepolia]: types_1.NetworkType.sepolia,
     [types_1.NetworkId.mainnet]: types_1.NetworkType.mainnet,
     [types_1.NetworkId['linea-goerli']]: types_1.NetworkType['linea-goerli'],
+    [types_1.NetworkId['linea-sepolia']]: types_1.NetworkType['linea-sepolia'],
     [types_1.NetworkId['linea-mainnet']]: types_1.NetworkType['linea-mainnet'],
 };
 //# sourceMappingURL=constants.js.map
\ No newline at end of file
diff --git a/node_modules/@metamask/controller-utils/dist/types.d.ts b/node_modules/@metamask/controller-utils/dist/types.d.ts
index 0356343..0958ace 100644
--- a/node_modules/@metamask/controller-utils/dist/types.d.ts
+++ b/node_modules/@metamask/controller-utils/dist/types.d.ts
@@ -18,6 +18,7 @@ export declare const NetworkType: {
     readonly goerli: "goerli";
     readonly sepolia: "sepolia";
     readonly 'linea-goerli': "linea-goerli";
+    readonly 'linea-sepolia': "linea-sepolia";
     readonly 'linea-mainnet': "linea-mainnet";
 };
 export declare type NetworkType = (typeof NetworkType)[keyof typeof NetworkType];
@@ -38,6 +39,7 @@ export declare enum BuiltInNetworkName {
     Goerli = "goerli",
     Sepolia = "sepolia",
     LineaGoerli = "linea-goerli",
+    LineaSepolia = "linea-sepolia",
     LineaMainnet = "linea-mainnet",
     Aurora = "aurora"
 }
@@ -52,6 +54,7 @@ export declare const ChainId: {
     readonly sepolia: "0xaa36a7";
     readonly aurora: "0x4e454152";
     readonly "linea-goerli": "0xe704";
+    readonly "linea-sepolia": "0xe705";
     readonly "linea-mainnet": "0xe708";
 };
 export declare type ChainId = (typeof ChainId)[keyof typeof ChainId];
@@ -63,6 +66,7 @@ export declare const NetworkId: {
     readonly goerli: "5";
     readonly sepolia: "11155111";
     readonly "linea-goerli": "59140";
+    readonly "linea-sepolia": "59141";
     readonly "linea-mainnet": "59144";
 };
 export declare type NetworkId = (typeof NetworkId)[keyof typeof NetworkId];
@@ -71,6 +75,7 @@ export declare enum NetworksTicker {
     goerli = "GoerliETH",
     sepolia = "SepoliaETH",
     'linea-goerli' = "LineaETH",
+    'linea-sepolia' = "LineaETH",
     'linea-mainnet' = "ETH",
     rpc = ""
 }
diff --git a/node_modules/@metamask/controller-utils/dist/types.js b/node_modules/@metamask/controller-utils/dist/types.js
index 90d517e..1cd28db 100644
--- a/node_modules/@metamask/controller-utils/dist/types.js
+++ b/node_modules/@metamask/controller-utils/dist/types.js
@@ -9,6 +9,7 @@ exports.InfuraNetworkType = {
     goerli: 'goerli',
     sepolia: 'sepolia',
     'linea-goerli': 'linea-goerli',
+    'linea-sepolia': 'linea-sepolia',
     'linea-mainnet': 'linea-mainnet',
 };
 /**
@@ -36,6 +37,7 @@ var BuiltInNetworkName;
     BuiltInNetworkName["Goerli"] = "goerli";
     BuiltInNetworkName["Sepolia"] = "sepolia";
     BuiltInNetworkName["LineaGoerli"] = "linea-goerli";
+    BuiltInNetworkName["LineaSepolia"] = "linea-sepolia";
     BuiltInNetworkName["LineaMainnet"] = "linea-mainnet";
     BuiltInNetworkName["Aurora"] = "aurora";
 })(BuiltInNetworkName = exports.BuiltInNetworkName || (exports.BuiltInNetworkName = {}));
@@ -50,6 +52,7 @@ exports.ChainId = {
     [BuiltInNetworkName.Sepolia]: '0xaa36a7',
     [BuiltInNetworkName.Aurora]: '0x4e454152',
     [BuiltInNetworkName.LineaGoerli]: '0xe704',
+    [BuiltInNetworkName.LineaSepolia]: '0xe705',
     [BuiltInNetworkName.LineaMainnet]: '0xe708', // toHex(59144)
 };
 /**
@@ -60,6 +63,7 @@ exports.NetworkId = {
     [exports.InfuraNetworkType.goerli]: '5',
     [exports.InfuraNetworkType.sepolia]: '11155111',
     [exports.InfuraNetworkType['linea-goerli']]: '59140',
+    [exports.InfuraNetworkType['linea-sepolia']]: '59141',
     [exports.InfuraNetworkType['linea-mainnet']]: '59144',
 };
 var NetworksTicker;
@@ -68,6 +72,7 @@ var NetworksTicker;
     NetworksTicker["goerli"] = "GoerliETH";
     NetworksTicker["sepolia"] = "SepoliaETH";
     NetworksTicker["linea-goerli"] = "LineaETH";
+    NetworksTicker["linea-sepolia"] = "LineaETH";
     NetworksTicker["linea-mainnet"] = "ETH";
     NetworksTicker["rpc"] = "";
 })(NetworksTicker = exports.NetworksTicker || (exports.NetworksTicker = {}));
