diff --git a/node_modules/@metamask/rpc-errors/dist/chunk-77LIU62I.js b/node_modules/@metamask/rpc-errors/dist/chunk-77LIU62I.js
index f7be896..534305f 100644
--- a/node_modules/@metamask/rpc-errors/dist/chunk-77LIU62I.js
+++ b/node_modules/@metamask/rpc-errors/dist/chunk-77LIU62I.js
@@ -35,9 +35,14 @@ var JsonRpcError = class extends Error {
         serialized.data.cause = _chunkLIUXO4DWjs.serializeCause.call(void 0, this.data.cause);
       }
     }
+    /**
+     * passing the stack creates a loop, possible conflicting with hermes, since it's hermes that throws an error
+     */
+    /* 
     if (this.stack) {
       serialized.stack = this.stack;
     } 
+    */
     return serialized;
   }
   /**
