diff --git a/node_modules/@metamask/smart-transactions-controller/dist/SmartTransactionsController.js b/node_modules/@metamask/smart-transactions-controller/dist/SmartTransactionsController.js
index cd9bd16..79fb857 100644
--- a/node_modules/@metamask/smart-transactions-controller/dist/SmartTransactionsController.js
+++ b/node_modules/@metamask/smart-transactions-controller/dist/SmartTransactionsController.js
@@ -77,8 +77,8 @@ class SmartTransactionsController extends polling_controller_1.StaticIntervalPol
         this.getMetaMetricsProps = getMetaMetricsProps;
         this.initializeSmartTransactionsForChainId();
         __classPrivateFieldGet(this, _SmartTransactionsController_instances, "m", _SmartTransactionsController_ensureUniqueSmartTransactions).call(this);
-        onNetworkStateChange(({ providerConfig: newProvider }) => {
-            const { chainId } = newProvider;
+        onNetworkStateChange((state) => {
+            const { chainId } = this.getNetworkClientById(state.selectedNetworkClientId);
             const isNewChainId = chainId !== this.config.chainId;
             this.configure({ chainId });
             this.initializeSmartTransactionsForChainId();
