diff --git a/node_modules/@metamask/utils/dist/chunk-QEPVHEP7.js b/node_modules/@metamask/utils/dist/chunk-QEPVHEP7.js
index b852707..b528bd2 100644
--- a/node_modules/@metamask/utils/dist/chunk-QEPVHEP7.js
+++ b/node_modules/@metamask/utils/dist/chunk-QEPVHEP7.js
@@ -232,6 +232,9 @@ function add0x(hexadecimal) {
   return `0x${hexadecimal}`;
 }
 function remove0x(hexadecimal) {
+  if (typeof hexadecimal !== 'string') {
+    return hexadecimal;
+  }
   if (hexadecimal.startsWith("0x") || hexadecimal.startsWith("0X")) {
     return hexadecimal.substring(2);
   }
