diff --git a/node_modules/@metamask/preferences-controller/dist/PreferencesController.d.ts b/node_modules/@metamask/preferences-controller/dist/PreferencesController.d.ts
index aae1f9d..9f4d064 100644
--- a/node_modules/@metamask/preferences-controller/dist/PreferencesController.d.ts
+++ b/node_modules/@metamask/preferences-controller/dist/PreferencesController.d.ts
@@ -61,9 +61,12 @@ export interface PreferencesState extends BaseState {
     useTokenDetection: boolean;
     useNftDetection: boolean;
     openSeaEnabled: boolean;
+    securityAlertsEnabled: boolean;
+    isMultiAccountBalancesEnabled: boolean;
     disabledRpcMethodPreferences: {
-        [methodName: string]: boolean;
+      [methodName: string]: boolean;
     };
+    showTestNetworks: boolean;  
 }
 /**
  * Controller that stores shared settings and exposes convenience methods
@@ -167,6 +170,12 @@ export declare class PreferencesController extends BaseController<BaseConfig, Pr
      * @param openSeaEnabled - Boolean indicating user preference on using OpenSea's API.
      */
     setOpenSeaEnabled(openSeaEnabled: boolean): void;
+    /**
+     * Toggle the security alert enabled setting.
+     *
+     * @param securityAlertsEnabled - Boolean indicating user preference on using security alerts.
+     */
+    setSecurityAlertsEnabled(securityAlertsEnabled): void;
     /**
      * A setter for the user preferences to enable/disable rpc methods.
      *
@@ -174,5 +183,17 @@ export declare class PreferencesController extends BaseController<BaseConfig, Pr
      * @param isEnabled - true to enable the rpc method, false to disable it.
      */
     setDisabledRpcMethodPreference(methodName: string, isEnabled: boolean): void;
+    /**
+     * A setter for the user preferences to enable/disable fetch of multiple accounts balance.
+     *
+     * @param isMultiAccountBalancesEnabled - true to enable multiple accounts balance fetch, false to fetch only selectedAddress.
+     */
+    setIsMultiAccountBalancesEnabled(isMultiAccountBalancesEnabled: boolean): void;
+    /**
+     * A setter for the user have the test networks visible/hidden.
+     *
+     * @param showTestNetworks - true to show test networks, false to hidden.
+     */
+    setShowTestNetworks(showTestNetworks: boolean): void;
 }
 export default PreferencesController;
diff --git a/node_modules/@metamask/preferences-controller/dist/PreferencesController.js b/node_modules/@metamask/preferences-controller/dist/PreferencesController.js
index 4ebc51c..a73b576 100644
--- a/node_modules/@metamask/preferences-controller/dist/PreferencesController.js
+++ b/node_modules/@metamask/preferences-controller/dist/PreferencesController.js
@@ -29,9 +29,12 @@ class PreferencesController extends base_controller_1.BaseController {
             useTokenDetection: true,
             useNftDetection: false,
             openSeaEnabled: false,
+            securityAlertsEnabled: false,
+            isMultiAccountBalancesEnabled: true,
             disabledRpcMethodPreferences: {
-                eth_sign: false,
+              eth_sign: false,
             },
+            showTestNetworks: false,      
         };
         this.initialize();
     }
@@ -240,6 +243,14 @@ class PreferencesController extends base_controller_1.BaseController {
             this.update({ useNftDetection: false });
         }
     }
+    /**
+     * Toggle the security alert enabled setting.
+     *
+     * @param securityAlertsEnabled - Boolean indicating user preference on using security alerts.
+     */
+    setSecurityAlertsEnabled(securityAlertsEnabled) {
+        this.update({ securityAlertsEnabled });
+    }
     /**
      * A setter for the user preferences to enable/disable rpc methods.
      *
@@ -251,6 +262,24 @@ class PreferencesController extends base_controller_1.BaseController {
         const newDisabledRpcMethods = Object.assign(Object.assign({}, disabledRpcMethodPreferences), { [methodName]: isEnabled });
         this.update({ disabledRpcMethodPreferences: newDisabledRpcMethods });
     }
+
+    /**
+     * A setter for the user preferences to enable/disable fetch of multiple accounts balance.
+     *
+     * @param isMultiAccountBalancesEnabled - true to enable multiple accounts balance fetch, false to fetch only selectedAddress.
+     */
+    setIsMultiAccountBalancesEnabled(isMultiAccountBalancesEnabled) {
+        this.update({ isMultiAccountBalancesEnabled });
+    }
+
+    /**
+     * A setter for the user have the test networks visible/hidden.
+     *
+     * @param showTestNetworks - true to show test networks, false to hidden.
+     */
+    setShowTestNetworks(showTestNetworks) {
+        this.update({ showTestNetworks });
+    }
 }
 exports.PreferencesController = PreferencesController;
 exports.default = PreferencesController;
