diff --git a/node_modules/@metamask/preferences-controller/dist/PreferencesController.d.ts b/node_modules/@metamask/preferences-controller/dist/PreferencesController.d.ts
index 323a8e8..0c5ed1b 100644
--- a/node_modules/@metamask/preferences-controller/dist/PreferencesController.d.ts
+++ b/node_modules/@metamask/preferences-controller/dist/PreferencesController.d.ts
@@ -35,7 +35,8 @@ export interface PreferencesState extends BaseState {
     selectedAddress: string;
     useTokenDetection: boolean;
     useNftDetection: boolean;
-    openSeaEnabled: boolean;
+    displayNftMedia: boolean;
+    useSafeChainsListValidation: boolean,
     securityAlertsEnabled: boolean;
     isMultiAccountBalancesEnabled: boolean;
     disabledRpcMethodPreferences: {
@@ -43,6 +44,8 @@ export interface PreferencesState extends BaseState {
     };
     showTestNetworks: boolean;
     isIpfsGatewayEnabled: boolean;
+    securityAlertsEnabled: boolean;
+    showIncomingTransactions: {[hexChainId: string]: boolean};
 }
 /**
  * Controller that stores shared settings and exposes convenience methods
@@ -125,11 +128,17 @@ export declare class PreferencesController extends BaseController<BaseConfig, Pr
      */
     setUseNftDetection(useNftDetection: boolean): void;
     /**
-     * Toggle the opensea enabled setting.
+     * Toggle the display nft media enabled setting.
+     *
+     * @param displayNftMedia - Boolean indicating user preference on using web2 third parties.
+     */
+    setDisplayNftMedia(displayNftMedia: boolean): void;
+    /**
+     * Toggle the use safe chains list validation.
      *
-     * @param openSeaEnabled - Boolean indicating user preference on using OpenSea's API.
+     * @param useSafeChainsListValidation - Boolean indicating user preference on using chainid.network third part to check safe networks.
      */
-    setOpenSeaEnabled(openSeaEnabled: boolean): void;
+    setUseSafeChainsListValidation(useSafeChainsListValidation: boolean): void;
     /**
      * Toggle the security alert enabled setting.
      *
@@ -161,6 +170,11 @@ export declare class PreferencesController extends BaseController<BaseConfig, Pr
      * @param isIpfsGatewayEnabled - true to enable ipfs source
      */
     setIsIpfsGatewayEnabled(isIpfsGatewayEnabled: boolean): void;
+    setEnableNetworkIncomingTransactions(
+        chainId: string,
+        isIncomingTransactionNetworkEnable: boolean,
+    ):void;
+
 }
 export default PreferencesController;
 //# sourceMappingURL=PreferencesController.d.ts.map
\ No newline at end of file
diff --git a/node_modules/@metamask/preferences-controller/dist/PreferencesController.js b/node_modules/@metamask/preferences-controller/dist/PreferencesController.js
index 1a98ab0..fc7ac7d 100644
--- a/node_modules/@metamask/preferences-controller/dist/PreferencesController.js
+++ b/node_modules/@metamask/preferences-controller/dist/PreferencesController.js
@@ -3,6 +3,31 @@ Object.defineProperty(exports, "__esModule", { value: true });
 exports.PreferencesController = void 0;
 const base_controller_1 = require("@metamask/base-controller");
 const controller_utils_1 = require("@metamask/controller-utils");
+const { deprecate } = require("util");
+
+const ETHERSCAN_SUPPORTED_CHAIN_IDS = {
+                MAINNET: '0x1',
+                GOERLI: '0x5',
+                BSC: '0x38',
+                BSC_TESTNET: '0x61',
+                OPTIMISM: '0xa',
+                OPTIMISM_SEPOLIA: '0xaa37dc',
+                POLYGON: '0x89',
+                POLYGON_TESTNET: '0x13881',
+                AVALANCHE: '0xa86a',
+                AVALANCHE_TESTNET: '0xa869',
+                FANTOM: '0xfa',
+                FANTOM_TESTNET: '0xfa2',
+                SEPOLIA: '0xaa36a7',
+                LINEA_GOERLI: '0xe704',
+                LINEA_SEPOLIA: '0xe705',
+                LINEA_MAINNET: '0xe708',
+                MOONBEAM: '0x504',
+                MOONBEAM_TESTNET: '0x507',
+                MOONRIVER: '0x505',
+                GNOSIS: '0x64',
+              };
+
 /**
  * Controller that stores shared settings and exposes convenience methods
  */
@@ -27,7 +52,8 @@ class PreferencesController extends base_controller_1.BaseController {
             selectedAddress: '',
             useTokenDetection: true,
             useNftDetection: false,
-            openSeaEnabled: false,
+            displayNftMedia: true,
+            useSafeChainsListValidation: true,
             securityAlertsEnabled: false,
             isMultiAccountBalancesEnabled: true,
             disabledRpcMethodPreferences: {
@@ -35,6 +61,28 @@ class PreferencesController extends base_controller_1.BaseController {
             },
             showTestNetworks: false,
             isIpfsGatewayEnabled: true,
+            showIncomingTransactions: {
+                                [ETHERSCAN_SUPPORTED_CHAIN_IDS.MAINNET]: true,
+                                [ETHERSCAN_SUPPORTED_CHAIN_IDS.GOERLI]: true,
+                                [ETHERSCAN_SUPPORTED_CHAIN_IDS.BSC]: true,
+                                [ETHERSCAN_SUPPORTED_CHAIN_IDS.BSC_TESTNET]: true,
+                                [ETHERSCAN_SUPPORTED_CHAIN_IDS.OPTIMISM]: true,
+                                [ETHERSCAN_SUPPORTED_CHAIN_IDS.OPTIMISM_SEPOLIA]: true,
+                                [ETHERSCAN_SUPPORTED_CHAIN_IDS.POLYGON]: true,
+                                [ETHERSCAN_SUPPORTED_CHAIN_IDS.POLYGON_TESTNET]: true,
+                                [ETHERSCAN_SUPPORTED_CHAIN_IDS.AVALANCHE]: true,
+                                [ETHERSCAN_SUPPORTED_CHAIN_IDS.AVALANCHE_TESTNET]: true,
+                                [ETHERSCAN_SUPPORTED_CHAIN_IDS.FANTOM]: true,
+                                [ETHERSCAN_SUPPORTED_CHAIN_IDS.FANTOM_TESTNET]: true,
+                                [ETHERSCAN_SUPPORTED_CHAIN_IDS.SEPOLIA]: true,
+                                [ETHERSCAN_SUPPORTED_CHAIN_IDS.LINEA_GOERLI]: true,
+                                [ETHERSCAN_SUPPORTED_CHAIN_IDS.LINEA_SEPOLIA]: true,
+                                [ETHERSCAN_SUPPORTED_CHAIN_IDS.LINEA_MAINNET]: true,
+                                [ETHERSCAN_SUPPORTED_CHAIN_IDS.MOONBEAM]: true,
+                                [ETHERSCAN_SUPPORTED_CHAIN_IDS.MOONBEAM_TESTNET]: true,
+                                [ETHERSCAN_SUPPORTED_CHAIN_IDS.MOONRIVER]: true,
+                                [ETHERSCAN_SUPPORTED_CHAIN_IDS.GNOSIS]: true,
+                              }
         };
         this.initialize();
     }
@@ -81,6 +129,7 @@ class PreferencesController extends base_controller_1.BaseController {
      *
      * @param address - Address of the identity to associate.
      * @param label - New label to assign.
+     * @deprecated use Engine.setAccountLabel instead.
      */
     setAccountLabel(address, label) {
         address = (0, controller_utils_1.toChecksumHexAddress)(address);
@@ -160,6 +209,9 @@ class PreferencesController extends base_controller_1.BaseController {
      *
      * @param selectedAddress - Ethereum address.
      */
+    /**
+    * @deprecated use Engine.setSelectedAddress instead.
+    */
     setSelectedAddress(selectedAddress) {
         this.update({ selectedAddress: (0, controller_utils_1.toChecksumHexAddress)(selectedAddress) });
     }
@@ -185,22 +237,33 @@ class PreferencesController extends base_controller_1.BaseController {
      * @param useNftDetection - Boolean indicating user preference on NFT detection.
      */
     setUseNftDetection(useNftDetection) {
-        if (useNftDetection && !this.state.openSeaEnabled) {
-            throw new Error('useNftDetection cannot be enabled if openSeaEnabled is false');
+        if (useNftDetection && !this.state.displayNftMedia) {
+            throw new Error('useNftDetection cannot be enabled if displayNftMedia is false');
         }
         this.update({ useNftDetection });
     }
     /**
-     * Toggle the opensea enabled setting.
+     * Toggle the display nft media enabled setting.
      *
-     * @param openSeaEnabled - Boolean indicating user preference on using OpenSea's API.
+     * @param displayNftMedia - Boolean indicating user preference on using web2 third parties.
      */
-    setOpenSeaEnabled(openSeaEnabled) {
-        this.update({ openSeaEnabled });
-        if (!openSeaEnabled) {
+    setDisplayNftMedia(displayNftMedia) {
+        this.update({ displayNftMedia });
+        if (!displayNftMedia) {
             this.update({ useNftDetection: false });
         }
     }
+    /**
+     * Toggle the use safe chains list validation.
+     *
+     * @param useSafeChainsListValidation - Boolean indicating user preference on using chainid.network third part to check safe networks.
+     */
+        setUseSafeChainsListValidation(useSafeChainsListValidation) {
+          this.update({ useSafeChainsListValidation });
+          if (!useSafeChainsListValidation) {
+              this.update({ useSafeChainsListValidation: false });
+          }
+      }
     /**
      * Toggle the security alert enabled setting.
      *
@@ -244,7 +307,25 @@ class PreferencesController extends base_controller_1.BaseController {
     setIsIpfsGatewayEnabled(isIpfsGatewayEnabled) {
         this.update({ isIpfsGatewayEnabled });
     }
+    /**
+     * A setter for the user allow to be fetched IPFS content
+     *
+     * @param chainId - On hexadecimal format to enable the incoming transaction network
+     * @param isIncomingTransactionNetworkEnable - true to enable incoming transactions
+     */
+    setEnableNetworkIncomingTransactions(
+            chainId,
+            isIncomingTransactionNetworkEnable,
+        ) {
+            this.update({
+              showIncomingTransactions: {
+                ...this.state.showIncomingTransactions,
+                [chainId]: isIncomingTransactionNetworkEnable,
+              },
+            });
+        }
 }
+exports.ETHERSCAN_SUPPORTED_CHAIN_IDS = ETHERSCAN_SUPPORTED_CHAIN_IDS;
 exports.PreferencesController = PreferencesController;
 exports.default = PreferencesController;
 //# sourceMappingURL=PreferencesController.js.map
\ No newline at end of file
