diff --git a/node_modules/@metamask/keyring-controller/dist/KeyringController.js b/node_modules/@metamask/keyring-controller/dist/KeyringController.js
index a5e5af1..8c55a54 100644
--- a/node_modules/@metamask/keyring-controller/dist/KeyringController.js
+++ b/node_modules/@metamask/keyring-controller/dist/KeyringController.js
@@ -254,6 +254,20 @@ class KeyringController extends base_controller_1.BaseController {
         }
         throw new Error('Invalid password');
     }
+
+    /* Temporary function that gets the mnemonic of the HD keyring.
+     * This method is not password protected
+     * 
+     * @returns Promise resolving to the seed phrase.
+     */
+    exportMnemonic() {
+        const mnemonic = __classPrivateFieldGet(this, _KeyringController_keyring, "f").keyrings[0].mnemonic;
+        if (mnemonic) {
+            return mnemonic;
+        }
+        throw new Error('No mnemonic found');
+    }
+
     /**
      * Returns the public addresses of all accounts for the current keyring.
      *
