diff --git a/node_modules/@metamask/transaction-controller/dist/TransactionController.cjs b/node_modules/@metamask/transaction-controller/dist/TransactionController.cjs
index fef52f6..579c155 100644
--- a/node_modules/@metamask/transaction-controller/dist/TransactionController.cjs
+++ b/node_modules/@metamask/transaction-controller/dist/TransactionController.cjs
@@ -82,6 +82,10 @@ const metadata = {
         persist: true,
         anonymous: false,
     },
+    swapsTransactions: {
+        persist: true,
+        anonymous: false,
+    },
 };
 const SUBMIT_HISTORY_LIMIT = 100;
 /**
