diff --git a/node_modules/@metamask/network-enablement-controller/dist/NetworkEnablementController.cjs b/node_modules/@metamask/network-enablement-controller/dist/NetworkEnablementController.cjs
index 9ddb3ce..7f0b262 100644
--- a/node_modules/@metamask/network-enablement-controller/dist/NetworkEnablementController.cjs
+++ b/node_modules/@metamask/network-enablement-controller/dist/NetworkEnablementController.cjs
@@ -207,9 +207,6 @@ class NetworkEnablementController extends base_controller_1.BaseController {
     disableNetwork(chainId) {
         const derivedKeys = (0, utils_2.deriveKeys)(chainId);
         const { namespace, storageKey } = derivedKeys;
-        if ((0, utils_2.isOnlyNetworkEnabledInNamespace)(this.state, derivedKeys)) {
-            throw new Error('Cannot disable the last remaining enabled network');
-        }
         this.update((s) => {
             s.enabledNetworkMap[namespace][storageKey] = false;
         });
