diff --git a/node_modules/@metamask/snaps-controllers/dist/snaps/SnapController.cjs b/node_modules/@metamask/snaps-controllers/dist/snaps/SnapController.cjs
index 908f4e7..8f66c3f 100644
--- a/node_modules/@metamask/snaps-controllers/dist/snaps/SnapController.cjs
+++ b/node_modules/@metamask/snaps-controllers/dist/snaps/SnapController.cjs
@@ -963,12 +963,7 @@ class SnapController extends base_controller_1.BaseController {
      */
     async clearState() {
         const snapIds = Object.keys(this.state.snaps);
-        if (this.#closeAllConnections) {
-            snapIds.forEach((snapId) => {
-                this.#closeAllConnections?.(snapId);
-            });
-        }
-        await this.messagingSystem.call('ExecutionService:terminateAllSnaps');
+        await this.stopAllSnaps();
         snapIds.forEach((snapId) => this.#revokeAllSnapPermissions(snapId));
         this.update((state) => {
             state.snaps = {};
