diff --git a/node_modules/@metamask/preferences-controller/dist/PreferencesController.d.cts b/node_modules/@metamask/preferences-controller/dist/PreferencesController.d.cts
index 1a6fce1..7c808cf 100644
--- a/node_modules/@metamask/preferences-controller/dist/PreferencesController.d.cts
+++ b/node_modules/@metamask/preferences-controller/dist/PreferencesController.d.cts
@@ -121,6 +121,18 @@ export type PreferencesState = {
      * Controls whether balance and assets are hidden or not
      */
     privacyMode: boolean;
+    /**
+     * Controls whether NFT media is displayed
+     */
+    displayNftMedia: boolean;
+    /**
+     * Controls token network filtering
+     */
+    tokenNetworkFilter: Record<string, string>;
+    /**
+     * Controls whether the multi RPC modal is shown
+     */
+    showMultiRpcModal: boolean;
 };
 declare const name = "PreferencesController";
 export type PreferencesControllerGetStateAction = ControllerGetStateAction<typeof name, PreferencesState>;
@@ -274,6 +286,24 @@ export declare class PreferencesController extends BaseController<typeof name, P
      * @param privacyMode - true to enable privacy mode, false to disable it.
      */
     setPrivacyMode(privacyMode: boolean): void;
+    /**
+     * Set the token network filter configuration setting.
+     *
+     * @param tokenNetworkFilter - Object mapping chain IDs to boolean values indicating which networks to filter tokens by.
+     */
+    setTokenNetworkFilter(tokenNetworkFilter: Record<string, boolean>): void;
+    /**
+     * A setter for the user preferences to enable/disable display of NFT media.
+     *
+     * @param displayNftMedia - true to enable display of NFT media, false to disable it.
+     */
+    setDisplayNftMedia(displayNftMedia: boolean): void;
+    /**
+     * A setter for the user preferences to enable/disable display of NFT media.
+     *
+     * @param displayNftMedia - true to enable display of NFT media, false to disable it.
+     */
+    setShowMultiRpcModal(showMultiRpcModal: boolean): void;
 }
 export default PreferencesController;
 //# sourceMappingURL=PreferencesController.d.cts.map
\ No newline at end of file
