diff --git a/node_modules/@metamask/network-controller/dist/NetworkController.cjs b/node_modules/@metamask/network-controller/dist/NetworkController.cjs
index e0bf9fc..2155395 100644
--- a/node_modules/@metamask/network-controller/dist/NetworkController.cjs
+++ b/node_modules/@metamask/network-controller/dist/NetworkController.cjs
@@ -110,6 +110,12 @@ const controllerName = 'NetworkController';
  */
 function getDefaultNetworkConfigurationsByChainId() {
     return Object.values(controller_utils_1.InfuraNetworkType).reduce((obj, infuraNetworkType) => {
+        if (
+            infuraNetworkType === controller_utils_1.InfuraNetworkType.goerli ||
+            infuraNetworkType === controller_utils_1.InfuraNetworkType['linea-goerli']
+        ) {
+            return obj;
+        }
         const chainId = controller_utils_1.ChainId[infuraNetworkType];
         const rpcEndpointUrl =
         // This ESLint rule mistakenly produces an error.
