diff --git a/node_modules/@metamask/message-manager/dist/MessageManager.js b/node_modules/@metamask/message-manager/dist/MessageManager.js
index e7c5d99..11948f9 100644
--- a/node_modules/@metamask/message-manager/dist/MessageManager.js
+++ b/node_modules/@metamask/message-manager/dist/MessageManager.js
@@ -58,6 +58,7 @@ class MessageManager extends AbstractMessageManager_1.AbstractMessageManager {
         const messageData = {
             id: messageId,
             messageParams,
+            securityAlertResponse: req?.securityAlertResponse,
             status: 'unapproved',
             time: Date.now(),
             type: 'eth_sign',
diff --git a/node_modules/@metamask/message-manager/dist/PersonalMessageManager.js b/node_modules/@metamask/message-manager/dist/PersonalMessageManager.js
index 036bd3b..c9bd117 100644
--- a/node_modules/@metamask/message-manager/dist/PersonalMessageManager.js
+++ b/node_modules/@metamask/message-manager/dist/PersonalMessageManager.js
@@ -58,6 +58,7 @@ class PersonalMessageManager extends AbstractMessageManager_1.AbstractMessageMan
         const messageData = {
             id: messageId,
             messageParams,
+            securityAlertResponse: req?.securityAlertResponse,
             status: 'unapproved',
             time: Date.now(),
             type: 'personal_sign',
diff --git a/node_modules/@metamask/message-manager/dist/TypedMessageManager.js b/node_modules/@metamask/message-manager/dist/TypedMessageManager.js
index e4b1aa3..f3abc5d 100644
--- a/node_modules/@metamask/message-manager/dist/TypedMessageManager.js
+++ b/node_modules/@metamask/message-manager/dist/TypedMessageManager.js
@@ -67,6 +67,7 @@ class TypedMessageManager extends AbstractMessageManager_1.AbstractMessageManage
         const messageData = {
             id: messageId,
             messageParams,
+            securityAlertResponse: req?.securityAlertResponse,
             status: 'unapproved',
             time: Date.now(),
             type: 'eth_signTypedData',
