diff --git a/node_modules/@metamask/permission-controller/dist/chunk-3X6WO7UG.js b/node_modules/@metamask/permission-controller/dist/chunk-3X6WO7UG.js
index e45cc3f..e37d09d 100644
--- a/node_modules/@metamask/permission-controller/dist/chunk-3X6WO7UG.js
+++ b/node_modules/@metamask/permission-controller/dist/chunk-3X6WO7UG.js
@@ -26,7 +26,11 @@ function invalidParams(opts) {
   });
 }
 function userRejectedRequest(data) {
-  return _rpcerrors.providerErrors.userRejectedRequest({ data });
+  if (data) {
+    return _rpcerrors.providerErrors.userRejectedRequest({ data });
+  } else {
+    throw new Error(`data is required for userRejectedRequest error. ${data} was provided.`);
+  }
 }
 function internalError(message, data) {
   return _rpcerrors.rpcErrors.internal({ message, data });
