diff --git a/node_modules/@metamask/keyring-controller/node_modules/@metamask/message-manager/dist/PersonalMessageManager.d.ts b/node_modules/@metamask/keyring-controller/node_modules/@metamask/message-manager/dist/PersonalMessageManager.d.ts
index 5bf0005..f372485 100644
--- a/node_modules/@metamask/keyring-controller/node_modules/@metamask/message-manager/dist/PersonalMessageManager.d.ts
+++ b/node_modules/@metamask/keyring-controller/node_modules/@metamask/message-manager/dist/PersonalMessageManager.d.ts
@@ -25,7 +25,10 @@ export interface PersonalMessage extends AbstractMessage {
  */
 export interface PersonalMessageParams extends AbstractMessageParams {
     data: string;
-    siwe?: SIWEMessage;
+    siwe?: {
+        isSIWEMessage: boolean;
+        parsedMessage: null;
+    };
 }
 /**
  * @type MessageParamsMetamask
