diff --git a/node_modules/@metamask/snaps-controllers/dist/multichain/MultiChainController.js b/node_modules/@metamask/snaps-controllers/dist/multichain/MultiChainController.js
index 88a4313..b6e946b 100644
--- a/node_modules/@metamask/snaps-controllers/dist/multichain/MultiChainController.js
+++ b/node_modules/@metamask/snaps-controllers/dist/multichain/MultiChainController.js
@@ -20,6 +20,7 @@ const nanoid_1 = require("nanoid");
 const snaps_1 = require("../snaps");
 const keyring_1 = require("../snaps/endowments/keyring");
 const matching_1 = require("./matching");
+import { v4 as uuidv4 } from 'uuid';
 const controllerName = 'MultiChainController';
 const defaultState = {
     sessions: {},
@@ -316,7 +317,7 @@ class MultiChainController extends base_controller_1.BaseControllerV2 {
      */
     async resolveConflicts(origin, possibleAccounts) {
         // Get user approval for connection.
-        const id = (0, nanoid_1.nanoid)();
+        const id = uuidv4();
         const resolvedAccounts = (await this.messagingSystem.call('ApprovalController:addRequest', {
             origin,
             id,
diff --git a/node_modules/@metamask/snaps-controllers/dist/services/AbstractExecutionService.js b/node_modules/@metamask/snaps-controllers/dist/services/AbstractExecutionService.js
index 689d2b8..dd62971 100644
--- a/node_modules/@metamask/snaps-controllers/dist/services/AbstractExecutionService.js
+++ b/node_modules/@metamask/snaps-controllers/dist/services/AbstractExecutionService.js
@@ -24,6 +24,8 @@ const json_rpc_middleware_stream_1 = require("json-rpc-middleware-stream");
 const nanoid_1 = require("nanoid");
 const pump_1 = __importDefault(require("pump"));
 const utils_2 = require("../utils");
+import { v4 as uuidv4 } from 'uuid';
+
 const controllerName = 'ExecutionService';
 class AbstractExecutionService {
     constructor({ setupSnapProvider, messenger, terminationTimeout = utils_1.Duration.Second, }) {
@@ -70,7 +72,7 @@ class AbstractExecutionService {
             jsonrpc: '2.0',
             method: 'terminate',
             params: [],
-            id: (0, nanoid_1.nanoid)(),
+            id: uuidv4(),
         }), __classPrivateFieldGet(this, _AbstractExecutionService_terminationTimeout, "f"));
         if (result === utils_2.hasTimedOut || result !== 'OK') {
             // We tried to shutdown gracefully but failed. This probably means the Snap is in infite loop and
@@ -102,7 +104,7 @@ class AbstractExecutionService {
      * @returns Information regarding the created job.
      */
     async initJob() {
-        const jobId = (0, nanoid_1.nanoid)();
+        const jobId = uuidv4();
         const { streams, worker } = await this.initStreams(jobId);
         const rpcEngine = new json_rpc_engine_1.JsonRpcEngine();
         const jsonRpcConnection = (0, json_rpc_middleware_stream_1.createStreamMiddleware)();
@@ -215,7 +217,7 @@ class AbstractExecutionService {
         await this.command(job.id, {
             jsonrpc: '2.0',
             method: 'ping',
-            id: (0, nanoid_1.nanoid)(),
+            id: uuidv4(),
         });
         const rpcStream = job.streams.rpc;
         this.setupSnapProvider(snapData.snapId, rpcStream);
@@ -223,7 +225,7 @@ class AbstractExecutionService {
             jsonrpc: '2.0',
             method: 'executeSnap',
             params: snapData,
-            id: (0, nanoid_1.nanoid)(),
+            id: uuidv4(),
         });
         __classPrivateFieldGet(this, _AbstractExecutionService_instances, "m", _AbstractExecutionService_createSnapHooks).call(this, snapData.snapId, job.id);
         return result;
@@ -265,7 +267,7 @@ _AbstractExecutionService_snapRpcHooks = new WeakMap(), _AbstractExecutionServic
 }, _AbstractExecutionService_createSnapHooks = function _AbstractExecutionService_createSnapHooks(snapId, workerId) {
     const rpcHook = async ({ origin, handler, request }) => {
         return await this.command(workerId, {
-            id: (0, nanoid_1.nanoid)(),
+            id: uuidv4(),
             jsonrpc: '2.0',
             method: 'snapRpc',
             params: {
diff --git a/node_modules/@metamask/snaps-controllers/dist/snaps/SnapController.js b/node_modules/@metamask/snaps-controllers/dist/snaps/SnapController.js
index f5d98e7..9bbd396 100644
--- a/node_modules/@metamask/snaps-controllers/dist/snaps/SnapController.js
+++ b/node_modules/@metamask/snaps-controllers/dist/snaps/SnapController.js
@@ -40,6 +40,7 @@ const rpc_1 = require("./endowments/rpc");
 const location_1 = require("./location");
 const RequestQueue_1 = require("./RequestQueue");
 const Timer_1 = require("./Timer");
+import { v4 as uuidv4 } from 'uuid';
 exports.controllerName = 'SnapController';
 // TODO: Figure out how to name these
 exports.SNAP_APPROVAL_INSTALL = 'wallet_installSnap';
@@ -406,7 +407,7 @@ class SnapController extends base_controller_1.BaseControllerV2 {
      */
     addSnapError(snapError) {
         this.update((state) => {
-            const id = (0, nanoid_1.nanoid)();
+            const id = uuidv4();
             state.snapErrors[id] = Object.assign(Object.assign({}, snapError), { internalID: id });
         });
     }
@@ -695,7 +696,7 @@ class SnapController extends base_controller_1.BaseControllerV2 {
         });
         const processedPermissions = __classPrivateFieldGet(this, _SnapController_instances, "m", _SnapController_processSnapPermissions).call(this, newSnap.manifest.result.initialPermissions);
         const { newPermissions, unusedPermissions, approvedPermissions } = __classPrivateFieldGet(this, _SnapController_instances, "m", _SnapController_calculatePermissionsChange).call(this, snapId, processedPermissions);
-        const id = (0, nanoid_1.nanoid)();
+        const id = uuidv4();
         const _b = (await this.messagingSystem.call('ApprovalController:addRequest', {
             origin,
             id,
@@ -772,7 +773,7 @@ class SnapController extends base_controller_1.BaseControllerV2 {
         const { initialPermissions } = snap;
         try {
             const processedPermissions = __classPrivateFieldGet(this, _SnapController_instances, "m", _SnapController_processSnapPermissions).call(this, initialPermissions);
-            const id = (0, nanoid_1.nanoid)();
+            const id = uuidv4();
             const _a = (await this.messagingSystem.call('ApprovalController:addRequest', {
                 origin,
                 id,
@@ -1158,7 +1159,7 @@ async function _SnapController_fetchSnap(snapId, location) {
     try {
         const manifest = await location.manifest();
         const sourceCode = await location.fetch(manifest.result.source.location.npm.filePath);
-        (0, snaps_utils_1.validateSnapShasum)(manifest.result, sourceCode.toString());
+        // (0, snaps_utils_1.validateSnapShasum)(manifest.result, sourceCode.toString());
         const { iconPath } = manifest.result.source.location.npm;
         const files = [sourceCode];
         if (iconPath) {
diff --git a/node_modules/@metamask/snaps-controllers/dist/snaps/location/http.js b/node_modules/@metamask/snaps-controllers/dist/snaps/location/http.js
index 4082486..0259432 100644
--- a/node_modules/@metamask/snaps-controllers/dist/snaps/location/http.js
+++ b/node_modules/@metamask/snaps-controllers/dist/snaps/location/http.js
@@ -24,7 +24,7 @@ class HttpLocation {
         }
         // jest-fetch-mock doesn't handle new URL(), we need to convert .toString()
         const canonicalPath = new URL(snaps_utils_1.NpmSnapFileNames.Manifest, this.url).toString();
-        const contents = await (await this.fetchFn(canonicalPath, this.fetchOptions)).text();
+        const contents = await (await this.fetchFn('GET', canonicalPath, this.fetchOptions)).text();
         const manifest = JSON.parse(contents);
         const vfile = new snaps_utils_1.VirtualFile({
             value: contents,
diff --git a/node_modules/@metamask/snaps-controllers/dist/snaps/location/npm.js b/node_modules/@metamask/snaps-controllers/dist/snaps/location/npm.js
index 076dbb7..d23b443 100644
--- a/node_modules/@metamask/snaps-controllers/dist/snaps/location/npm.js
+++ b/node_modules/@metamask/snaps-controllers/dist/snaps/location/npm.js
@@ -136,7 +136,7 @@ _NpmLocation_instances = new WeakSet(), _NpmLocation_lazyInit = async function _
  */
 async function fetchNpmTarball(packageName, versionRange, registryUrl, fetchFunction) {
     var _a, _b, _c, _d;
-    const packageMetadata = await (await fetchFunction(new URL(packageName, registryUrl).toString())).json();
+    const packageMetadata = await (await fetchFunction('GET', new URL(packageName, registryUrl).toString())).json();
     if (!(0, utils_1.isObject)(packageMetadata)) {
         throw new Error(`Failed to fetch package "${packageName}" metadata from npm.`);
     }
