diff --git a/node_modules/@metamask/keyring-controller/node_modules/@metamask/message-manager/dist/PersonalMessageManager.d.ts b/node_modules/@metamask/keyring-controller/node_modules/@metamask/message-manager/dist/PersonalMessageManager.d.ts
index 5bf0005..2702abd 100644
--- a/node_modules/@metamask/keyring-controller/node_modules/@metamask/message-manager/dist/PersonalMessageManager.d.ts
+++ b/node_modules/@metamask/keyring-controller/node_modules/@metamask/message-manager/dist/PersonalMessageManager.d.ts
@@ -25,7 +25,31 @@ export interface PersonalMessage extends AbstractMessage {
  */
 export interface PersonalMessageParams extends AbstractMessageParams {
     data: string;
-    siwe?: SIWEMessage;
+    /**
+     * ============================== PATCH INFORMATION ==============================
+     * This patch addresses a type error that was introduced in
+     * @metamask+message-manager+7.0.1.patch. The error is triggered when a message is
+     * not a valid SIWE message. The patch modifies the data type as follows,
+     * 
+     * Previous data type:
+     * ```ts
+     * siwe?: SIWEMessage;
+     * ```
+     * 
+     * Current data type introduced in @metamask+message-manager+7.0.1.patch:
+     * ```ts
+     * siwe?: {
+     *   isSIWEMessage: boolean;
+     *   parsedMessage: null;
+     * };
+     * ```
+     * 
+     * ===============================================================================
+     */
+    siwe?: {
+        isSIWEMessage: boolean;
+        parsedMessage: null;
+    };
 }
 /**
  * @type MessageParamsMetamask
