diff --git a/node_modules/@metamask/phishing-controller/dist/PhishingController.js b/node_modules/@metamask/phishing-controller/dist/PhishingController.js
index 5919d93..b87c013 100644
--- a/node_modules/@metamask/phishing-controller/dist/PhishingController.js
+++ b/node_modules/@metamask/phishing-controller/dist/PhishingController.js
@@ -108,12 +108,18 @@ class PhishingController extends base_controller_1.BaseController {
         this.defaultState = {
             phishingLists: [
                 {
-                    allowlist: config_json_1.default.whitelist,
-                    blocklist: config_json_1.default.blacklist,
-                    fuzzylist: config_json_1.default.fuzzylist,
-                    tolerance: config_json_1.default.tolerance,
-                    version: config_json_1.default.version,
-                    name: ListNames.MetaMask,
+                    // allowlist: config_json_1.default.whitelist,
+                    // blocklist: config_json_1.default.blacklist,
+                    // fuzzylist: config_json_1.default.fuzzylist,
+                    // tolerance: config_json_1.default.tolerance,
+                    // version: config_json_1.default.version,
+                    // name: ListNames.MetaMask,
+                    allowlist: [],
+                    blocklist: [],
+                    fuzzylist: [],
+                    tolerance: [],
+                    version: [],
+                    name: [],
                     lastUpdated: 0,
                 },
             ],
@@ -122,6 +128,10 @@ class PhishingController extends base_controller_1.BaseController {
             stalelistLastFetched: 0,
         };
         this.initialize();
+        // this.updatePhishingDetector();
+    }
+
+    start() {
         this.updatePhishingDetector();
     }
     /**
