diff --git a/node_modules/@metamask/controller-utils/dist/util.d.cts b/node_modules/@metamask/controller-utils/dist/util.d.cts
index f7e651b..ee30b81 100644
--- a/node_modules/@metamask/controller-utils/dist/util.d.cts
+++ b/node_modules/@metamask/controller-utils/dist/util.d.cts
@@ -114,7 +114,7 @@ export declare function safelyExecuteWithTimeout<Result>(operation: () => Promis
  * @param address - The address to convert.
  * @returns The address in 0x-prefixed hexadecimal checksummed form if it is valid.
  */
-export declare function toChecksumHexAddress(address: string): string;
+export declare function toChecksumHexAddress(address: string): Hex;
 /**
  * Convert an address to a checksummed hexadecimal address.
  *
