diff --git a/node_modules/@react-native-community/blur/android/build.gradle b/node_modules/@react-native-community/blur/android/build.gradle
index acb9a88..ad12554 100644
--- a/node_modules/@react-native-community/blur/android/build.gradle
+++ b/node_modules/@react-native-community/blur/android/build.gradle
@@ -60,6 +60,10 @@ android {
       }
     }
   }
+
+  buildFeatures {
+    buildConfig true
+  }
 }
 
 repositories {
