diff --git a/node_modules/@segment/analytics-react-native/src/analytics.ts b/node_modules/@segment/analytics-react-native/src/analytics.ts
index fb75134..c612cfd 100644
--- a/node_modules/@segment/analytics-react-native/src/analytics.ts
+++ b/node_modules/@segment/analytics-react-native/src/analytics.ts
@@ -481,10 +481,12 @@ export class SegmentClient {
     this.logger.info('SCREEN event saved', event);
   }
 
-  async track(eventName: string, options?: JsonMap) {
+  async track(eventName: string, options?: JsonMap, userId?: string, anonymousId?: string) {
     const event = createTrackEvent({
       event: eventName,
       properties: options,
+      userId: userId,
+      anonymousId: anonymousId,
     });
 
     await this.process(event);
@@ -647,11 +649,8 @@ export class SegmentClient {
     this.appState = nextAppState;
   }
 
-  reset(resetAnonymousId: boolean = true) {
-    const anonymousId =
-      resetAnonymousId === true
-        ? getUUID()
-        : this.store.userInfo.get().anonymousId;
+  reset() {
+    const anonymousId = '0x0000000000000000';
 
     this.store.userInfo.set({
       anonymousId,
diff --git a/node_modules/@segment/analytics-react-native/src/client.tsx b/node_modules/@segment/analytics-react-native/src/client.tsx
index 3309986..da1109e 100644
--- a/node_modules/@segment/analytics-react-native/src/client.tsx
+++ b/node_modules/@segment/analytics-react-native/src/client.tsx
@@ -67,6 +67,6 @@ export const useAnalytics = (): ClientMethods => {
     flush: async () => client.flush(),
     group: async (...args) => client.group(...args),
     alias: async (...args) => client.alias(...args),
-    reset: async (...args) => client.reset(...args),
+    reset: async () => client.reset(),
   };
 };
diff --git a/node_modules/@segment/analytics-react-native/src/events.ts b/node_modules/@segment/analytics-react-native/src/events.ts
index d940f68..cbfd29e 100644
--- a/node_modules/@segment/analytics-react-native/src/events.ts
+++ b/node_modules/@segment/analytics-react-native/src/events.ts
@@ -17,13 +17,19 @@ import {
 export const createTrackEvent = ({
   event,
   properties = {},
+  userId,
+  anonymousId,
 }: {
   event: string;
   properties?: JsonMap;
+  userId?: string;
+  anonymousId?: string;
 }): TrackEventType => ({
   type: EventType.TrackEvent,
   event,
   properties,
+  userId,
+  anonymousId,
 });
 
 export const createScreenEvent = ({
@@ -89,13 +95,8 @@ export const applyRawEventData = (
 ): SegmentEvent => {
   return {
     ...event,
-    anonymousId: userInfo.anonymousId,
     messageId: getUUID(),
     timestamp: new Date().toISOString(),
     integrations: event.integrations ?? {},
-    userId:
-      isAliasEvent(event) || isIdentifyEvent(event)
-        ? event.userId
-        : userInfo.userId,
   };
 };
