diff --git a/node_modules/@sentry/core/build/types/exports.d.ts b/node_modules/@sentry/core/build/types/exports.d.ts
index 5402c50..79a162b 100644
--- a/node_modules/@sentry/core/build/types/exports.d.ts
+++ b/node_modules/@sentry/core/build/types/exports.d.ts
@@ -6,8 +6,10 @@ import type { ExclusiveEventHintOrCaptureContext } from './utils/prepareEvent';
  * @param exception The exception to capture.
  * @param hint Optional additional data to attach to the Sentry event.
  * @returns the id of the captured Sentry event.
+ * 
+ * Patch to enforce better stack tracing using Error - https://docs.sentry.io/platforms/javascript/usage/#capturing-errors
  */
-export declare function captureException(exception: unknown, hint?: ExclusiveEventHintOrCaptureContext): string;
+export declare function captureException(exception: Error, hint?: ExclusiveEventHintOrCaptureContext): string;
 /**
  * Captures a message event and sends it to Sentry.
  *
