diff --git a/node_modules/appwright/dist/providers/browserstack/index.js b/node_modules/appwright/dist/providers/browserstack/index.js
index 638c5df..71b3d65 100644
--- a/node_modules/appwright/dist/providers/browserstack/index.js
+++ b/node_modules/appwright/dist/providers/browserstack/index.js
@@ -242,6 +242,9 @@ class BrowserStackDeviceProvider {
                 "bstack:options": {
                     debug: true,
                     interactiveDebugging: true,
+                    local: process.env.BROWSERSTACK_LOCAL || false, // needed to run e2e on CI
+                    localIdentifier: process.env.GITHUB_RUN_ID, // needed to run e2e on CI
+                    appProfiling: true, // needed to profile the app
                     networkLogs: true,
                     appiumVersion: "2.6.0",
                     enableCameraImageInjection: this.project.use.device?.enableCameraImageInjection,
diff --git a/node_modules/appwright/dist/providers/emulator/index.js b/node_modules/appwright/dist/providers/emulator/index.js
index f3bc150..e45fd2a 100644
--- a/node_modules/appwright/dist/providers/emulator/index.js
+++ b/node_modules/appwright/dist/providers/emulator/index.js
@@ -75,7 +75,7 @@ Follow the steps mentioned in ${androidSimulatorConfigDocLink} to run test on An
                 "appium:autoAcceptAlerts": true,
                 "appium:fullReset": true,
                 "appium:deviceOrientation": this.project.use.device?.orientation,
-                "appium:settings[snapshotMaxDepth]": 62,
+                "appium:settings[snapshotMaxDepth]": 80,
                 "appium:wdaLaunchTimeout": 300_000,
             },
         };
