diff --git a/node_modules/appwright/dist/config.js b/node_modules/appwright/dist/config.js
index a51913b..8c0335c 100644
--- a/node_modules/appwright/dist/config.js
+++ b/node_modules/appwright/dist/config.js
@@ -25,7 +25,7 @@ const defaultConfig = {
     fullyParallel: false,
     forbidOnly: false,
     retries: process.env.CI ? 2 : 0,
-    workers: 2,
+    workers: 1,
     reporter: [["list"], ["html", { open: "always" }]],
     use: {
         // TODO: Use this for actions
diff --git a/node_modules/appwright/dist/providers/browserstack/index.js b/node_modules/appwright/dist/providers/browserstack/index.js
index 638c5df..1733feb 100644
--- a/node_modules/appwright/dist/providers/browserstack/index.js
+++ b/node_modules/appwright/dist/providers/browserstack/index.js
@@ -242,6 +242,9 @@ class BrowserStackDeviceProvider {
                 "bstack:options": {
                     debug: true,
                     interactiveDebugging: true,
+                    local: process.env.BROWSERSTACK_LOCAL || false, // needed to run e2e on CI
+                    localIdentifier: process.env.GITHUB_RUN_ID, // needed to run e2e on CI
+                    appProfiling: true, // needed to profile the app
                     networkLogs: true,
                     appiumVersion: "2.6.0",
                     enableCameraImageInjection: this.project.use.device?.enableCameraImageInjection,
@@ -250,16 +253,17 @@ class BrowserStackDeviceProvider {
                     osVersion: this.project.use.device.osVersion,
                     platformName: platformName,
                     deviceOrientation: this.project.use.device?.orientation,
-                    buildName: `${projectName} ${platformName}`,
+                    buildName: process.env.BROWSERSTACK_BUILD_NAME || `${projectName} ${platformName}`,
                     sessionName: `${projectName} ${platformName} test`,
                     buildIdentifier: process.env.GITHUB_ACTIONS === "true"
-                        ? `CI ${process.env.GITHUB_RUN_ID}`
+                        ? ""
                         : process.env.USER,
                 },
                 "appium:autoGrantPermissions": true,
