diff --git a/node_modules/asyncstorage-down/default-opts.js b/node_modules/asyncstorage-down/default-opts.js
index c076d3d..d5f7705 100644
--- a/node_modules/asyncstorage-down/default-opts.js
+++ b/node_modules/asyncstorage-down/default-opts.js
@@ -1,5 +1,5 @@
 module.exports = {
   get AsyncStorage() {
-    return require('react-native').AsyncStorage
+    return require('@react-native-async-storage/async-storage').default
   }
 }
