diff --git a/node_modules/detox/src/utils/ExclusiveLockfile.js b/node_modules/detox/src/utils/ExclusiveLockfile.js
index da29ae9..45ad4a3 100644
--- a/node_modules/detox/src/utils/ExclusiveLockfile.js
+++ b/node_modules/detox/src/utils/ExclusiveLockfile.js
@@ -107,7 +107,7 @@ class ExclusiveLockfile {
     this._ensureFileExists();
 
     await retry(this._options.retry, () => {
-      const operationResult = plockfile.lockSync(this._lockFilePath);
+      const operationResult = plockfile.lockSync(this._lockFilePath, { stale: 20000 });
 
       this._isLocked = true;
       this._invalidate();
