diff --git a/node_modules/eth-json-rpc-infura/src/index.js b/node_modules/eth-json-rpc-infura/src/index.js
index fa299ea..80740d6 100644
--- a/node_modules/eth-json-rpc-infura/src/index.js
+++ b/node_modules/eth-json-rpc-infura/src/index.js
@@ -104,30 +104,15 @@ async function performFetch(network, req, res, source){
 function fetchConfigFromReq({ network, req, source }) {
   const requestOrigin = req.origin || 'internal'
   const cleanReq = normalizeReq(req)
-  const { method, params } = cleanReq
-
   const fetchParams = {}
-  let fetchUrl = `https://api.infura.io/v1/jsonrpc/${network}`
-  const isPostMethod = POST_METHODS.includes(method)
-  if (isPostMethod) {
-    fetchParams.method = 'POST'
-    fetchParams.headers = {
-      'Accept': 'application/json',
-      'Content-Type': 'application/json',
-      ...(source ? {'Infura-Source': `${source}/${requestOrigin}`} : {}),
-    },
-    fetchParams.body = JSON.stringify(cleanReq)
-  } else {
-    fetchParams.method = 'GET'
-    if (source) {
-      fetchParams.headers = {
-        'Infura-Source': `${source}/${requestOrigin}`
-      }
-    }
-    const paramsString = encodeURIComponent(JSON.stringify(params))
-    fetchUrl += `/${method}?params=${paramsString}`
+  let fetchUrl = `https://${network}.infura.io/v3/${process.env.MM_INFURA_KEY}`
+  fetchParams.method = 'POST'
+  fetchParams.headers = {
+    'Accept': 'application/json',
+    'Content-Type': 'application/json'
   }
-
+  if (source) fetchParams.headers['Infura-Source'] = `${source}/${requestOrigin}`
+  fetchParams.body = JSON.stringify(cleanReq)
   return { fetchUrl, fetchParams }
 }
 
