diff --git a/node_modules/expo-modules-core/ios/Core/ExpoBridgeModule.h b/node_modules/expo-modules-core/ios/Core/ExpoBridgeModule.h
index 067e470..3a6a879 100644
--- a/node_modules/expo-modules-core/ios/Core/ExpoBridgeModule.h
+++ b/node_modules/expo-modules-core/ios/Core/ExpoBridgeModule.h
@@ -1,6 +1,6 @@
 // Copyright 2024-present 650 Industries. All rights reserved.
 
-#import <React/React-Core-umbrella.h>
+#import <React/React-Core.common-umbrella.h>
 #import <ExpoModulesCore/EXNativeModulesProxy.h>
 #import <ExpoModulesCore/EXModuleRegistry.h>
 
diff --git a/node_modules/expo-modules-core/ios/EXAppDefines.m b/node_modules/expo-modules-core/ios/EXAppDefines.m
index 87c1651..e12e335 100644
--- a/node_modules/expo-modules-core/ios/EXAppDefines.m
+++ b/node_modules/expo-modules-core/ios/EXAppDefines.m
@@ -1,7 +1,7 @@
 // Copyright 2016-present 650 Industries. All rights reserved.
 
 #import <ExpoModulesCore/EXAppDefines.h>
-#import <React/React-Core-umbrella.h>
+#import <React/React-Core.common-umbrella.h>
 
 @implementation EXAppDefines
 
diff --git a/node_modules/expo-modules-core/ios/Fabric/ExpoFabricViewObjC.h b/node_modules/expo-modules-core/ios/Fabric/ExpoFabricViewObjC.h
index 75ed964..2549c3e 100644
--- a/node_modules/expo-modules-core/ios/Fabric/ExpoFabricViewObjC.h
+++ b/node_modules/expo-modules-core/ios/Fabric/ExpoFabricViewObjC.h
@@ -4,7 +4,7 @@
 
 #ifdef RCT_NEW_ARCH_ENABLED
 
-#import <React/React-Core-umbrella.h>
+#import <React/React-Core.common-umbrella.h>
 
 #ifdef __cplusplus
 
diff --git a/node_modules/expo-modules-core/ios/Fabric/ExpoFabricViewObjC.mm b/node_modules/expo-modules-core/ios/Fabric/ExpoFabricViewObjC.mm
index 2eacddf..6833b28 100644
--- a/node_modules/expo-modules-core/ios/Fabric/ExpoFabricViewObjC.mm
+++ b/node_modules/expo-modules-core/ios/Fabric/ExpoFabricViewObjC.mm
@@ -10,7 +10,7 @@
 #import <ExpoModulesCore/ExpoViewComponentDescriptor.h>
 #import <ExpoModulesCore/Swift.h>
 
-#import <React/React-Core-umbrella.h>
+#import <React/React-Core.common-umbrella.h>
 
 #import <string.h>
 
diff --git a/node_modules/expo-modules-core/ios/JSI/EXJSIConversions.h b/node_modules/expo-modules-core/ios/JSI/EXJSIConversions.h
index 37f5fd6..a025bc4 100644
--- a/node_modules/expo-modules-core/ios/JSI/EXJSIConversions.h
+++ b/node_modules/expo-modules-core/ios/JSI/EXJSIConversions.h
@@ -5,7 +5,7 @@
 #import <Foundation/Foundation.h>
 
 #import <jsi/jsi.h>
-#import <React/React-Core-umbrella.h>
+#import <React/React-Core.common-umbrella.h>
 
 using namespace facebook;
 using namespace react;
diff --git a/node_modules/expo-modules-core/ios/JSI/EXJSIInstaller.h b/node_modules/expo-modules-core/ios/JSI/EXJSIInstaller.h
index b26eb1c..44fdb35 100644
--- a/node_modules/expo-modules-core/ios/JSI/EXJSIInstaller.h
+++ b/node_modules/expo-modules-core/ios/JSI/EXJSIInstaller.h
@@ -1,6 +1,6 @@
 // Copyright 2018-present 650 Industries. All rights reserved.
 
-#import <React/React-Core-umbrella.h>
+#import <React/React-Core.common-umbrella.h>
 
 // Swift classes need forward-declaration in the headers.
 @class EXAppContext;
diff --git a/node_modules/expo-modules-core/ios/JSI/EXJSIUtils.h b/node_modules/expo-modules-core/ios/JSI/EXJSIUtils.h
index 69785d1..f562539 100644
--- a/node_modules/expo-modules-core/ios/JSI/EXJSIUtils.h
+++ b/node_modules/expo-modules-core/ios/JSI/EXJSIUtils.h
@@ -6,7 +6,7 @@
 
 #import <jsi/jsi.h>
 #import <React/RCTBridgeModule.h>
-#import <React/React-Core-umbrella.h>
+#import <React/React-Core.common-umbrella.h>
 #import <ReactCommon/TurboModuleUtils.h>
 #import <ExpoModulesCore/ObjectDeallocator.h>
 
diff --git a/node_modules/expo-modules-core/ios/JSI/EXJSIUtils.mm b/node_modules/expo-modules-core/ios/JSI/EXJSIUtils.mm
index 7175b42..e73e49d 100644
--- a/node_modules/expo-modules-core/ios/JSI/EXJSIUtils.mm
+++ b/node_modules/expo-modules-core/ios/JSI/EXJSIUtils.mm
@@ -2,7 +2,7 @@
 
 #import <sstream>
 
-#import <React/React-Core-umbrella.h>
+#import <React/React-Core.common-umbrella.h>
 #import <ExpoModulesCore/EXJSIConversions.h>
 #import <ExpoModulesCore/EXJSIUtils.h>
 #import <ExpoModulesCore/JSIUtils.h>
diff --git a/node_modules/expo-modules-core/ios/JSI/EXJavaScriptRuntime.h b/node_modules/expo-modules-core/ios/JSI/EXJavaScriptRuntime.h
index 57047ce..da90d40 100644
--- a/node_modules/expo-modules-core/ios/JSI/EXJavaScriptRuntime.h
+++ b/node_modules/expo-modules-core/ios/JSI/EXJavaScriptRuntime.h
@@ -3,7 +3,7 @@
 #import <Foundation/Foundation.h>
 #import <ExpoModulesCore/EXJavaScriptValue.h>
 #import <ExpoModulesCore/EXJavaScriptObject.h>
-#import <React/React-Core-umbrella.h>
+#import <React/React-Core.common-umbrella.h>
 
 #ifdef __cplusplus
 
diff --git a/node_modules/expo-modules-core/ios/Legacy/EXBridgeModule.h b/node_modules/expo-modules-core/ios/Legacy/EXBridgeModule.h
index c720d91..22da308 100644
--- a/node_modules/expo-modules-core/ios/Legacy/EXBridgeModule.h
+++ b/node_modules/expo-modules-core/ios/Legacy/EXBridgeModule.h
@@ -1,6 +1,6 @@
 // Copyright 2018-present 650 Industries. All rights reserved.
 
-#import <React/React-Core-umbrella.h>
+#import <React/React-Core.common-umbrella.h>
 
 // Escape hatch for modules that both have to depend on React Native
 // and want to be exported as an internal universal module.
diff --git a/node_modules/expo-modules-core/ios/Legacy/ModuleRegistryAdapter/EXModuleRegistryAdapter.h b/node_modules/expo-modules-core/ios/Legacy/ModuleRegistryAdapter/EXModuleRegistryAdapter.h
index 516c8be..f11f790 100644
--- a/node_modules/expo-modules-core/ios/Legacy/ModuleRegistryAdapter/EXModuleRegistryAdapter.h
+++ b/node_modules/expo-modules-core/ios/Legacy/ModuleRegistryAdapter/EXModuleRegistryAdapter.h
@@ -1,6 +1,6 @@
 // Copyright 2018-present 650 Industries. All rights reserved.
 
-#import <React/React-Core-umbrella.h>
+#import <React/React-Core.common-umbrella.h>
 #import <ExpoModulesCore/EXModuleRegistryProvider.h>
 
 // An "adapter" over module registry, for given RCTBridge and NSString
diff --git a/node_modules/expo-modules-core/ios/Legacy/ModuleRegistryAdapter/EXModuleRegistryHolderReactModule.h b/node_modules/expo-modules-core/ios/Legacy/ModuleRegistryAdapter/EXModuleRegistryHolderReactModule.h
index 7ced47a..c07f972 100644
--- a/node_modules/expo-modules-core/ios/Legacy/ModuleRegistryAdapter/EXModuleRegistryHolderReactModule.h
+++ b/node_modules/expo-modules-core/ios/Legacy/ModuleRegistryAdapter/EXModuleRegistryHolderReactModule.h
@@ -1,6 +1,6 @@
 // Copyright 2018-present 650 Industries. All rights reserved.
 
-#import <React/React-Core-umbrella.h>
+#import <React/React-Core.common-umbrella.h>
 
 #import <ExpoModulesCore/EXModuleRegistry.h>
 
diff --git a/node_modules/expo-modules-core/ios/Legacy/NativeModulesProxy/EXNativeModulesProxy.h b/node_modules/expo-modules-core/ios/Legacy/NativeModulesProxy/EXNativeModulesProxy.h
index c65d952..d95c6db 100644
--- a/node_modules/expo-modules-core/ios/Legacy/NativeModulesProxy/EXNativeModulesProxy.h
+++ b/node_modules/expo-modules-core/ios/Legacy/NativeModulesProxy/EXNativeModulesProxy.h
@@ -1,6 +1,6 @@
 // Copyright 2018-present 650 Industries. All rights reserved.
 
-#import <React/React-Core-umbrella.h>
+#import <React/React-Core.common-umbrella.h>
 
 #import <ExpoModulesCore/EXInternalModule.h>
 #import <ExpoModulesCore/EXModuleRegistry.h>
diff --git a/node_modules/expo-modules-core/ios/Legacy/NativeModulesProxy/EXNativeModulesProxy.mm b/node_modules/expo-modules-core/ios/Legacy/NativeModulesProxy/EXNativeModulesProxy.mm
index be78486..1c00d04 100644
--- a/node_modules/expo-modules-core/ios/Legacy/NativeModulesProxy/EXNativeModulesProxy.mm
+++ b/node_modules/expo-modules-core/ios/Legacy/NativeModulesProxy/EXNativeModulesProxy.mm
@@ -2,7 +2,7 @@
 
 #import <objc/runtime.h>
 
-#import <React/React-Core-umbrella.h>
+#import <React/React-Core.common-umbrella.h>
 #import <React/RCTComponentViewFactory.h> // Allows non-umbrella since it's coming from React-RCTFabric
 
 #import <jsi/jsi.h>
diff --git a/node_modules/expo-modules-core/ios/Legacy/Services/EXReactLogHandler.m b/node_modules/expo-modules-core/ios/Legacy/Services/EXReactLogHandler.m
index bd176e5..1fe33ca 100644
--- a/node_modules/expo-modules-core/ios/Legacy/Services/EXReactLogHandler.m
+++ b/node_modules/expo-modules-core/ios/Legacy/Services/EXReactLogHandler.m
@@ -1,6 +1,6 @@
 // Copyright 2018-present 650 Industries. All rights reserved.
 
-#import <React/React-Core-umbrella.h>
+#import <React/React-Core.common-umbrella.h>
 
 #import <ExpoModulesCore/EXReactLogHandler.h>
 #import <ExpoModulesCore/EXDefines.h>
diff --git a/node_modules/expo-modules-core/ios/Legacy/Services/EXReactNativeAdapter.h b/node_modules/expo-modules-core/ios/Legacy/Services/EXReactNativeAdapter.h
index 8cc9565..078bcf5 100644
--- a/node_modules/expo-modules-core/ios/Legacy/Services/EXReactNativeAdapter.h
+++ b/node_modules/expo-modules-core/ios/Legacy/Services/EXReactNativeAdapter.h
@@ -1,6 +1,6 @@
 // Copyright 2018-present 650 Industries. All rights reserved.
 
-#import <React/React-Core-umbrella.h>
+#import <React/React-Core.common-umbrella.h>
 
 #import <ExpoModulesCore/EXUIManager.h>
 #import <ExpoModulesCore/EXInternalModule.h>
diff --git a/node_modules/expo-modules-core/ios/Legacy/Services/EXReactNativeAdapter.mm b/node_modules/expo-modules-core/ios/Legacy/Services/EXReactNativeAdapter.mm
index e033fa8..c4f12be 100644
--- a/node_modules/expo-modules-core/ios/Legacy/Services/EXReactNativeAdapter.mm
+++ b/node_modules/expo-modules-core/ios/Legacy/Services/EXReactNativeAdapter.mm
@@ -2,7 +2,7 @@
 
 #import <JavaScriptCore/JavaScriptCore.h>
 
-#import <React/React-Core-umbrella.h>
+#import <React/React-Core.common-umbrella.h>
 #import <ExpoModulesCore/EXReactNativeAdapter.h>
 #import <ExpoModulesCore/ExpoFabricViewObjC.h>
 
diff --git a/node_modules/expo-modules-core/ios/Legacy/Services/EXReactNativeEventEmitter.h b/node_modules/expo-modules-core/ios/Legacy/Services/EXReactNativeEventEmitter.h
index fdde7a0..717f570 100644
--- a/node_modules/expo-modules-core/ios/Legacy/Services/EXReactNativeEventEmitter.h
+++ b/node_modules/expo-modules-core/ios/Legacy/Services/EXReactNativeEventEmitter.h
@@ -1,6 +1,6 @@
 // Copyright 2018-present 650 Industries. All rights reserved.
 
-#import <React/React-Core-umbrella.h>
+#import <React/React-Core.common-umbrella.h>
 
 #import <ExpoModulesCore/EXInternalModule.h>
 #import <ExpoModulesCore/EXEventEmitterService.h>
diff --git a/node_modules/expo-modules-core/ios/RCTComponentData+Privates.h b/node_modules/expo-modules-core/ios/RCTComponentData+Privates.h
index a03803c..549f20e 100644
--- a/node_modules/expo-modules-core/ios/RCTComponentData+Privates.h
+++ b/node_modules/expo-modules-core/ios/RCTComponentData+Privates.h
@@ -1,6 +1,6 @@
 // Copyright 2021-present 650 Industries. All rights reserved.
 
-#import <React/React-Core-umbrella.h>
+#import <React/React-Core.common-umbrella.h>
 
 typedef void (^RCTPropBlockAlias)(id<RCTComponent> _Nonnull view, id _Nullable json);
 
diff --git a/node_modules/expo-modules-core/ios/RCTComponentData+Privates.m b/node_modules/expo-modules-core/ios/RCTComponentData+Privates.m
index 563c6e4..7cf2545 100644
--- a/node_modules/expo-modules-core/ios/RCTComponentData+Privates.m
+++ b/node_modules/expo-modules-core/ios/RCTComponentData+Privates.m
@@ -1,7 +1,7 @@
 // Copyright 2021-present 650 Industries. All rights reserved.
 
 #import <ExpoModulesCore/RCTComponentData+Privates.h>
-#import <React/React-Core-umbrella.h>
+#import <React/React-Core.common-umbrella.h>
 
 @implementation RCTComponentDataSwiftAdapter
 
