diff --git a/node_modules/react-native-modal/dist/modal.js b/node_modules/react-native-modal/dist/modal.js
index ec3cc0b..6471cfb 100644
--- a/node_modules/react-native-modal/dist/modal.js
+++ b/node_modules/react-native-modal/dist/modal.js
@@ -448,15 +448,15 @@ export class ReactNativeModal extends React.Component {
         if (this.props.onSwipe) {
             console.warn('`<Modal onSwipe="..." />` is deprecated and will be removed starting from 13.0.0. Use `<Modal onSwipeComplete="..." />` instead.');
         }
-        DeviceEventEmitter.addListener('didUpdateDimensions', this.handleDimensionsUpdate);
+        this._didUpdateDimensionsSubscription = DeviceEventEmitter.addListener('didUpdateDimensions', this.handleDimensionsUpdate);
         if (this.state.isVisible) {
             this.open();
         }
-        BackHandler.addEventListener('hardwareBackPress', this.onBackButtonPress);
+        this._hardwareBackPressSubscription = BackHandler.addEventListener('hardwareBackPress', this.onBackButtonPress);
     }
     componentWillUnmount() {
-        BackHandler.removeEventListener('hardwareBackPress', this.onBackButtonPress);
-        DeviceEventEmitter.removeListener('didUpdateDimensions', this.handleDimensionsUpdate);
+        this._didUpdateDimensionsSubscription.remove();
+        this._hardwareBackPressSubscription.remove();
         if (this.interactionHandle) {
             InteractionManager.clearInteractionHandle(this.interactionHandle);
             this.interactionHandle = null;
