diff --git a/node_modules/react-native-os/android/build.gradle b/node_modules/react-native-os/android/build.gradle
index 7055b60..784681d 100644
--- a/node_modules/react-native-os/android/build.gradle
+++ b/node_modules/react-native-os/android/build.gradle
@@ -44,5 +44,5 @@ repositories {
 }
 
 dependencies {
-    compile 'com.facebook.react:react-native:+'
+    implementation 'com.facebook.react:react-native:+'
 }
