diff --git a/node_modules/react-native/scripts/cocoapods/autolinking.rb b/node_modules/react-native/scripts/cocoapods/autolinking.rb
index 1111111..2222222 100644
--- a/node_modules/react-native/scripts/cocoapods/autolinking.rb
+++ b/node_modules/react-native/scripts/cocoapods/autolinking.rb
@@ -107,8 +107,6 @@ end
 #   - :ios_project_root_path - Absolute path to the react_native project's ios folder, e.g. /Users/foobar/project/rn_project/ios
 #   - :react_native_path - Relative path to the react_native from the project, e.g. ./node_modules/react-native
 def link_native_modules!(config)
-  Pod::UI.puts "link_native_modules! #{config}"
-
   if !(
     config[:ios_packages].is_a? Array and
     config[:ios_project_root_path].is_a? String and
