diff --git a/node_modules/react-native-aes-crypto-forked/android/build.gradle b/node_modules/react-native-aes-crypto-forked/android/build.gradle
index e717caf..583178f 100644
--- a/node_modules/react-native-aes-crypto-forked/android/build.gradle
+++ b/node_modules/react-native-aes-crypto-forked/android/build.gradle
@@ -5,7 +5,7 @@ android {
     buildToolsVersion "26.0.1"
 
     defaultConfig {
-        minSdkVersion 19
+        minSdkVersion 21
         targetSdkVersion 22
         versionCode 1
         versionName "1.0"
