diff --git a/node_modules/react-native/ReactAndroid/build.gradle b/node_modules/react-native/ReactAndroid/build.gradle
index edadad2..2cadd4f 100644
--- a/node_modules/react-native/ReactAndroid/build.gradle
+++ b/node_modules/react-native/ReactAndroid/build.gradle
@@ -440,15 +440,11 @@ task installArchives {
 }
 
 android {
-    compileSdkVersion 30
-    ndkVersion ANDROID_NDK_VERSION
-    if (ANDROID_NDK_PATH != null) {
-        ndkPath ANDROID_NDK_PATH
-    }
+    compileSdkVersion rootProject.ext.compileSdkVersion
 
     defaultConfig {
-        minSdkVersion(21)
-        targetSdkVersion(28)
+        minSdkVersion rootProject.ext.minSdkVersion
+		targetSdkVersion rootProject.ext.targetSdkVersion
         versionCode(1)
         versionName("1.0")
 
diff --git a/node_modules/react-native/ReactAndroid/gradle.properties b/node_modules/react-native/ReactAndroid/gradle.properties
index 857a818..c465984 100644
--- a/node_modules/react-native/ReactAndroid/gradle.properties
+++ b/node_modules/react-native/ReactAndroid/gradle.properties
@@ -13,7 +13,7 @@ JUNIT_VERSION=4.12
 ANDROIDX_TEST_VERSION=1.1.0
 FRESCO_VERSION=2.5.0
 OKHTTP_VERSION=4.9.1
-SO_LOADER_VERSION=0.10.1
+SO_LOADER_VERSION=0.10.4
 
 BOOST_VERSION=1_63_0
 DOUBLE_CONVERSION_VERSION=1.1.6
