diff --git a/node_modules/react-native-fabric/android/build.gradle b/node_modules/react-native-fabric/android/build.gradle
index 053ecfc..697b267 100644
--- a/node_modules/react-native-fabric/android/build.gradle
+++ b/node_modules/react-native-fabric/android/build.gradle
@@ -42,8 +42,8 @@ repositories {
 }

 dependencies {
-    compile "com.facebook.react:react-native:${safeExtGet('reactNative', '+')}"
-    compile('com.crashlytics.sdk.android:crashlytics:2.9.2@aar') {
+    api "com.facebook.react:react-native:${safeExtGet('reactNative', '+')}"
+    api('com.crashlytics.sdk.android:crashlytics:2.9.2@aar') {
         transitive = true;
     }
 }
